<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Introduction to Proteomics: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/uom/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicons/uom/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicons/uom/favicon-16x16.png">
<link rel="manifest" href="favicons/uom/site.webmanifest">
<link rel="mask-icon" href="favicons/uom/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="assets/images/uom-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="assets/images/uom-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Introduction to Proteomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Introduction to Proteomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="contact.html">Contact</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Introduction to Proteomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01dataset.html">1. Exploring the dataset</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02diann.html">2. Processing data with DIA-NN</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="contact.html">Contact</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01dataset"><p>Content from <a href="01dataset.html">Exploring the dataset</a></p>
<hr>
<p>Last updated on 2025-11-11 |

        <a href="https://example.com/template-source/lesson-example/edit/main/episodes/01dataset.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the data we are using in this tutorial?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the dataset we are using in this tutorial</li>
<li>Learn how to download publicly available proteomics data from within
R</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="the-data">The data<a class="anchor" aria-label="anchor" href="#the-data"></a>
</h2>
<hr class="half-width">
<p>The data we are using in this tutorial comes from a study done
exploring a non-invasive diagnostic approach for Inflammatory Bowel
Disease (IBD) using mass spectrometry and machine learning to analyse
the stool proteome. Traditional diagnostic methods like colonoscopies
are invasive, prompting the need for alternatives. The researchers
collected 123 stool samples and identified 48 differentially expressed
proteins, narrowing them down to 7 key proteins using feature selection.
A Support Vector Machine (SVM) model was developed, achieving 96%
sensitivity and 76% specificity in distinguishing active IBD patients
from symptomatic non-IBD patients. This approach demonstrates the
potential for accurate, non-invasive IBD diagnosis, improving patient
management and reducing the need for invasive procedures.</p>
<p><strong>For the purpose of this tutorial</strong>, we will be using a
subset of the total data presented in the paper, including: - 10 control
samples (6 from Batch One + 4 from Batch Three) - 10 aCD (active Crohn’s
disease) samples (5 from Batch One + 5 from Batch 3)</p>
<p><strong>Reference</strong> Shajari, E.; Gagné, D.; Malick, M.; Roy,
P.; Noël, J.-F.; Gagnon, H.; Brunet, M.A.; Delisle, M.; Boisvert, F.-M.;
Beaulieu, J.-F. Application of SWATH Mass Spectrometry and Machine
Learning in the Diagnosis of Inflammatory Bowel Disease Based on the
Stool Proteome. Biomedicines 2024, 12, 333. <a href="https://doi.org/10.3390/biomedicines12020333" class="external-link uri">https://doi.org/10.3390/biomedicines12020333</a>.</p>
<div id="discussion1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Can you see where to find the publicly available data in the
paper?</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h2>
<hr class="half-width">
<p>Let’s start by loading the packages required to access data stored on
PRIDE.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Learn more about PRIDE </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>PRIDE is part of the <strong>ProteomeXchange Consortium</strong>,
which provides open access to proteomics data.</p>
<p>For more background on navigating PRIDE data, you can look through
this course: <a href="https://www.ebi.ac.uk/training/online/courses/pride-quick-tour/" class="external-link">PRIDE
Quick Tour (EMBL-EBI Training)</a></p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">rpx</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>rpx</code> package gives us programmatic access to the
ProteomeXchange without having to leave R.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set the dataset identifier and load</span></span>
<span><span class="va">px_id</span> <span class="op">&lt;-</span> <span class="st">'PXD047585'</span> </span>
<span><span class="va">px</span> <span class="op">&lt;-</span> <span class="fu">PXDataset</span><span class="op">(</span><span class="va">px_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the dataset title</span></span>
<span><span class="fu">pxtitle</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Application of SWATH Mass Spectrometry and Machine Learning in Diagnosis of Inflammatory Bowel Disease Based on Stool Proteome"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print a link to the dataset on PRIDE</span></span>
<span><span class="fu">pxurl</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print the taxonomy (organism) information for the dataset</span></span>
<span><span class="fu">pxtax</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Homo sapiens (human)"</code></pre>
</div>
<div class="section level3">
<h3 id="exploring-dataset-files">Exploring dataset files<a class="anchor" aria-label="anchor" href="#exploring-dataset-files"></a>
</h3>
<p>Let’s see which files are included in this dataset.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Retrieve a list of dataset files</span></span>
<span><span class="va">px_files</span> <span class="op">&lt;-</span> <span class="fu">pxfiles</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the first few files</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">px_files</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "20201016_UdSjfb_20190115_freshstool_003.mzML"
[2] "20201016_UdSjfb_20190115_freshstool_003.wiff"
[3] "20201016_UdSjfb_20190115_freshstool_003.wiff.scan"
[4] "20201016_UdSjfb_20190115_freshstool_005.mzML"
[5] "20201016_UdSjfb_20190115_freshstool_005.wiff"
[6] "20201016_UdSjfb_20190115_freshstool_005.wiff.scan"</code></pre>
</div>
<p>PRIDE datasets often include a mix of raw data, search results,
processed outputs, and metadata.</p>
<p>To understand the data composition, we’ll examine file
extensions.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of files by extension</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="fu">sapply</span><span class="op">(</span><span class="fu">strsplit</span><span class="op">(</span><span class="va">px_files</span>, <span class="st">"\\."</span><span class="op">)</span>, <span class="va">tail</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
    fas    mzML     pdf    scan speclib     tsv     txt    wiff    xlsx
      1      78       1      78       2      13       1      78       3 </code></pre>
</div>
<p>Alternatively, we can get the same result with the tools package:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tools</span><span class="op">)</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="fu">file_ext</span><span class="op">(</span><span class="va">px_files</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
    fas    mzML     pdf    scan speclib     tsv     txt    wiff    xlsx
      1      78       1      78       2      13       1      78       3 </code></pre>
</div>
<p>Each file extension represents a specific proteomics data type. For
example, <code>.raw</code> may indicate vendor instrument output and
<code>.mzML</code> indicates an open format for MS data.</p>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Learn more about proteomics file formats </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler2" aria-labelledby="headingSpoiler2">
<div class="accordion-body">
<p>For more information about different proteomics file formats, check
out:</p>
<ul>
<li><p><a href="https://www.ebi.ac.uk/pride/markdownpage/pridefileformats" class="external-link">PRIDE
File Formats Documentation</a></p></li>
<li><p><a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3518119/" class="external-link">Paper
on mass spec file formats (PMC3518119)</a></p></li>
</ul>
</div>
</div>
</div>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Try visiting the dataset page on the PRIDE website and compare:</p>
<ul>
<li><p>The number of files listed there versus those returned by
rpx.</p></li>
<li><p>The types of files (e.g., .raw, .mzML, .txt, etc.)</p></li>
</ul>
<p>Questions to consider:</p>
<ul>
<li><p>Why might the PRIDE web interface and R output differ?</p></li>
<li><p>What are the advantages of using programmatic access via
rpx?</p></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>UPDATE</p>
<p>Programmatic access allows you to automate each step of your
analysis, making your workflow fully reproducible.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="downloading-required-files">Downloading required files<a class="anchor" aria-label="anchor" href="#downloading-required-files"></a>
</h3>
<p>Good news! We have already downloaded and processed the data for you,
as described in the next lesson. For future reference, you can use the
<strong>Curl</strong> package to download files as demonstrated
below.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">curl</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">curl</span><span class="fu">::</span><span class="fu">curl_download</span><span class="op">(</span>url<span class="op">=</span><span class="st">'https://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585/SampleAnnotation.xlsx'</span>,destfile <span class="op">=</span> <span class="st">'data/SampleAnnotation.xlsx'</span><span class="op">)</span></span></code></pre>
</div>
<ul>
<li>We retrieved the url in an earlier step using pxurl(), replacing <a href="ftp://" class="uri">ftp://</a> at the start with <a href="https://" class="external-link uri">https://</a>
</li>
<li>Specify the name of the file you wish to download, from the list we
sourced earlier using pxfiles()</li>
<li>Set the name and location for your downloaded file</li>
</ul>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>This dataset includes stool samples used to investigate non-invasive
diagnostic alternatives for Inflammatory Bowel Disease.</li>
<li>We can use the <em>rpx</em> package to download and explore data
from PRIDE or the ProteomeXchange Consortium without leaving R.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-02diann"><p>Content from <a href="02diann.html">Processing data with DIA-NN</a></p>
<hr>
<p>Last updated on 2025-11-11 |

        <a href="https://example.com/template-source/lesson-example/edit/main/episodes/02diann.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I process my output files from a mass spectrometry-based
proteomics experiment?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to process proteomics data using DIA-NN</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="processing-proteomics-data">Processing proteomics data<a class="anchor" aria-label="anchor" href="#processing-proteomics-data"></a>
</h2>
<hr class="half-width">
<p>There are many software options for processing mass
spectrometry-based proteomics output. These software <em>match mass
spectra</em> against an empirical or predicted spectral library to
<em>identify peptides</em> and infer the protein to which they
belong.</p>
<p>Some software you may have heard of include Spectronaut, Fragpipe,
MaxQuant and DIA-NN. Each software have slightly different functionality
and attributes, and are frequently updated. In this tutorial, we are
demonstrating the use of <strong>DIA-NN</strong>, a software designed to
process <strong>DIA</strong> proteomics data.</p>
<figure><img src="fig/02diann_peptideMatching.png" alt="Proteomics software like DIA-NN match observed spectra against a spectral library to identify peptides and assign them to a protein or protein group. Source: Galaxy Training Network" class="figure mx-auto d-block"><div class="figcaption">Proteomics software like DIA-NN match observed
spectra against a spectral library to identify peptides and assign them
to a protein or protein group. Source: <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/introduction/slides.html#15" class="external-link">Galaxy
Training Network</a>
</div>
</figure><div class="section level3">
<h3 id="why-dia-nn">Why DIA-NN?<a class="anchor" aria-label="anchor" href="#why-dia-nn"></a>
</h3>
<p>Key features of DIA-NN software include:</p>
<ul>
<li>
<strong>Neural network–based scoring:</strong> Improves
identification confidence.</li>
<li>
<strong>Interference correction:</strong> Enhances quantification
accuracy.</li>
<li>
<strong>Library-free analysis:</strong> Can work without a spectral
library (generating one directly from DIA data).</li>
<li>
<strong>High speed and efficiency:</strong> Suitable for large-scale
datasets.</li>
<li>
<strong>Cross-run normalization:</strong> Ensures consistent
quantification across experiments.</li>
<li>
<strong>Command-line executable:</strong> DIA-NN can be run from a
graphical user interface (GUI) or directly from the command line, making
it easy to integrate into an automated pipeline.</li>
</ul>
<p>DIA-NN is free to download. Detailed operating instructions and a
link to install the latest version of DIA-NN can be found on the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">GitHub
page</a>.</p>
<div id="important-note" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="important-note" class="callout-inner">
<h3 class="callout-title">Important Note</h3>
<div class="callout-content">
<p>Due to the amount of time it will take to process the data, you do
not need to install or operate DIA-NN for the purposes of this tutorial.
Pre-processed data is provided, which has been prepared following the
procedures described below.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="dia-nn-workflow">DIA-NN workflow<a class="anchor" aria-label="anchor" href="#dia-nn-workflow"></a>
</h2>
<hr class="half-width"></section><section><h2 class="section-heading" id="part-1-creating-a-spectral-library">Part 1: Creating a spectral library<a class="anchor" aria-label="anchor" href="#part-1-creating-a-spectral-library"></a>
</h2>
<hr class="half-width">
<p>DIA-NN analysis consists of two steps. The first step is to generate
a predicted spectral library for your experimental organism.</p>
<ol style="list-style-type: decimal"><li>
</li></ol>
<p>Variable mods</p>
<figure><img src="fig/02diann_peptideMatching.png" alt="Example of the DIA-NN GUI used to generate a predicted spectral library. Settings which have been altered from the default are highlighted in red" class="figure mx-auto d-block"><div class="figcaption">Example of the DIA-NN GUI used to generate a
predicted spectral library. Settings which have been altered from the
default are highlighted in red</div>
</figure><div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">C:\DIA-NN\2.2.0\diann.exe</span> <span class="kw">`</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">--lib</span> <span class="st">""</span> <span class="kw">`</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">--out</span> <span class="st">"C:\path\to\output\report.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with file path</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">--out-lib</span> <span class="st">"C:\path\to\output\report-lib.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with file path</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\UniprotFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\ContaminantsFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">--threads</span> 30 <span class="kw">`</span> <span class="co"># Replace with the number of threads to use for this analysis</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">--verbose</span> 1 <span class="kw">`</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">--qvalue</span> 0.01 <span class="kw">`</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="ex">--matrices</span>  <span class="kw">`</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="ex">--gen-spec-lib</span> <span class="kw">`</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="ex">--predictor</span> <span class="kw">`</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="ex">--reannotate</span> <span class="kw">`</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="ex">--fasta-search</span> <span class="kw">`</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="ex">--min-fr-mz</span> 200 <span class="kw">`</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="ex">--max-fr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="ex">--met-excision</span> <span class="kw">`</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="ex">--min-pep-len</span> 7 <span class="kw">`</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="ex">--max-pep-len</span> 30 <span class="kw">`</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="ex">--min-pr-mz</span> 300 <span class="kw">`</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="ex">--max-pr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="ex">--min-pr-charge</span> 1 <span class="kw">`</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="ex">--max-pr-charge</span> 4 <span class="kw">`</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="ex">--cut</span> K<span class="pp">*</span>,R<span class="pp">*</span> <span class="kw">`</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="ex">--missed-cleavages</span> 1 <span class="kw">`</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="ex">--unimod4</span> <span class="kw">`</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="ex">--rt-profiling</span> </span></code></pre>
</div>
<p>This library can be re-used for all of analyses of samples from the
same organism. It is recommended to re-download the UniProt fasta and
re-run this step approximately once a month to ensure your library is
up-to-date.</p>
</section><section><h2 class="section-heading" id="general-workflow">General Workflow<a class="anchor" aria-label="anchor" href="#general-workflow"></a>
</h2>
<hr class="half-width">
<p>The DIA-NN workflow typically consists of the following steps:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Prepare input data</strong>
<ul>
<li>Raw DIA data files (<code>.raw</code>, <code>.wiff</code>, or
converted <code>.mzML</code> files).</li>
<li>(Optional) A spectral library (<code>.tsv</code> or
<code>.speclib</code>).</li>
</ul>
</li>
<li>
<strong>Run DIA-NN</strong>
<ul>
<li>Can be run via the <strong>graphical user interface (GUI)</strong>
or <strong>command line</strong>.</li>
<li>Specify inputs, output directory, and parameters (e.g., FDR cutoff,
protein grouping, library mode).</li>
</ul>
</li>
<li>
<strong>Output files</strong>
<ul>
<li>Quantification tables for proteins and peptides
(<code>report.tsv</code> or <code>.parquet</code>).</li>
<li>Run summaries and performance metrics.</li>
<li>Optional normalized intensity data for downstream statistical
analysis.</li>
</ul>
</li>
<li>
<strong>Post-processing</strong>
<ul>
<li>Further analysis can be done in R (e.g., <code>tidyverse</code>,
<code>MSstats</code>, <code>limma</code>), Python, or visualization
tools.</li>
<li>Data can also be imported into platforms like
<strong>Perseus</strong>, <strong>Spectronaut</strong>, or
<strong>Skyline</strong> for downstream exploration.</li>
</ul>
</li>
</ol></section><section><h2 class="section-heading" id="example-command-line-usage">Example Command-Line Usage<a class="anchor" aria-label="anchor" href="#example-command-line-usage"></a>
</h2>
<hr class="half-width">
<p>Below is an example of running DIA-NN in command-line mode (adjust
paths for your setup):</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">DIA-NN.exe</span> <span class="at">--f</span> <span class="st">"path/to/mzML_files/"</span> <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>           <span class="at">--out</span> <span class="st">"path/to/output_folder/"</span> <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>           <span class="at">--lib</span> <span class="st">"path/to/library.tsv"</span> <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>           <span class="at">--threads</span> 8 <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>           <span class="at">--fasta</span> <span class="st">"path/to/fasta_file.fasta"</span> <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>           <span class="at">--qvalue</span> 0.01 <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>           <span class="at">--verbose</span> 1</span></code></pre>
</div>
<p>This command tells DIA-NN to:</p>
<p>Analyze all .mzML files in a folder</p>
<p>Use a specified spectral library and FASTA file</p>
<p>Run on 8 threads</p>
<p>Output results with a 1% false discovery rate cutoff</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  More about DIANN
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<ul>
<li><p>For more background information, see the original
publication:<br><em>Demichev, V., et al. (2020). DIA-NN: neural networks and
interference correction enable deep proteome coverage in high
throughput. Nature Methods.</em> <a href="https://doi.org/10.1038/s41592-019-0638-x" class="external-link">https://doi.org/10.1038/s41592-019-0638-x</a></p></li>
<li><p>For further details about the different DIA-NN settings, see the
[GitHub page] (<a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link uri">https://github.com/vdemichev/DiaNN?tab=readme-ov-file</a>)</p></li>
<li><p>If you come across any issues, the creators are fairly active at
responding to <a href="https://github.com/vdemichev/DiaNN/issues" class="external-link">issues</a> and <a href="https://github.com/vdemichev/DiaNN/discussions" class="external-link">discussions</a> on
their GitHub</p></li>
</ul>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>DIANN is fun and is theoretically easy to use!</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://example.com/template-source/lesson-example/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://example.com/template-source/lesson-example/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://example.com/template-source/lesson-example/" class="external-link">Source</a></p>
        <p>
        <a href="citation.html">Cite</a>
        | <a href="mailto:bioinformatics-training@unimelb.edu.au">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors.</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7)</a></p>
		</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "proteomics, mass spectrometry, DIA, DIA-NN, limpa, differential expression analysis, enrichment analysis, network analysis",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "dateCreated": "2025-10-30",
  "dateModified": "2025-11-11",
  "datePublished": "2025-11-11"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

