<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Introduction to Proteomics: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/uom/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicons/uom/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicons/uom/favicon-16x16.png">
<link rel="manifest" href="favicons/uom/site.webmanifest">
<link rel="mask-icon" href="favicons/uom/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="assets/images/uom-logo.svg"><span class="badge text-bg-info">
          <abbr title="This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-circle" style="border-radius: 5px"></i>
              Beta
            </a>
            <span class="visually-hidden">This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="assets/images/uom-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Introduction to Proteomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Introduction to Proteomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
<li><a class="dropdown-item" href="contact.html">Contact</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Introduction to Proteomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01dataset.html">1. Exploring the dataset</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02diann.html">2. Processing data with DIA-NN</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03datacleaning.html">3. Data cleaning with R</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04DEA.html">4. Differential expression analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05EnrichmentAnalysis.html">5. Network and Enrichment Analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Glossary</a></li>
<li><a href="contact.html">Contact</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01dataset"><p>Content from <a href="01dataset.html">Exploring the dataset</a></p>
<hr>
<p>Last updated on 2025-11-14 |

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/episodes/01dataset.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the data we are using in this tutorial?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the dataset we are using in this tutorial.</li>
<li>Learn how to download publicly available proteomics data from within
R.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="the-data">The data<a class="anchor" aria-label="anchor" href="#the-data"></a>
</h2>
<hr class="half-width">
<p>The data we are using in this tutorial comes from a study done
exploring a non-invasive diagnostic approach for Inflammatory Bowel
Disease (IBD) using mass spectrometry and machine learning to analyse
the stool proteome. Traditional diagnostic methods like colonoscopies
are invasive, prompting the need for alternatives. The researchers
collected 123 stool samples and identified 48 differentially expressed
proteins, narrowing them down to 7 key proteins using feature selection.
A Support Vector Machine (SVM) model was developed, achieving 96%
sensitivity and 76% specificity in distinguishing active IBD patients
from symptomatic non-IBD patients. This approach demonstrates the
potential for accurate, non-invasive IBD diagnosis, improving patient
management and reducing the need for invasive procedures.</p>
<p><strong>For the purpose of this tutorial</strong>, we will be using a
<strong>subset</strong> of the total data presented in the paper,
including:</p>
<ul>
<li>
<p>11 control (Ctrl) samples</p>
<ul>
<li>6 from batch one + 5 from batch three</li>
</ul>
</li>
<li>
<p>9 active Crohn’s disease (aCD) samples</p>
<ul>
<li>5 from batch one + 4 from batch three</li>
</ul>
</li>
</ul>
<p><strong>Reference:</strong></p>
<p>Shajari, E.; Gagné, D.; Malick, M.; Roy, P.; Noël, J.-F.; Gagnon, H.;
Brunet, M.A.; Delisle, M.; Boisvert, F.-M.; Beaulieu, J.-F. Application
of SWATH Mass Spectrometry and Machine Learning in the Diagnosis of
Inflammatory Bowel Disease Based on the Stool Proteome. Biomedicines
2024, 12, 333. <a href="https://doi.org/10.3390/biomedicines12020333" class="external-link uri">https://doi.org/10.3390/biomedicines12020333</a></p>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Can you see where to find the publicly available data in the
paper?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Publicly available data can be found under the ‘Data Availability
Statement’ in the paper.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h2>
<hr class="half-width">
<p>Let’s start by loading the packages required to access data stored on
PRIDE.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Learn more about PRIDE </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>PRIDE is part of the <strong>ProteomeXchange Consortium</strong>,
which provides open access to proteomics data.</p>
<p>For more background on navigating PRIDE data, you can look through
this course: <a href="https://www.ebi.ac.uk/training/online/courses/pride-quick-tour/" class="external-link">PRIDE
Quick Tour (EMBL-EBI Training)</a></p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">rpx</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>rpx</code> package gives us programmatic access to the
ProteomeXchange without having to leave R.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set the dataset identifier and load</span></span>
<span><span class="va">px_id</span> <span class="op">&lt;-</span> <span class="st">'PXD047585'</span> </span>
<span><span class="va">px</span> <span class="op">&lt;-</span> <span class="fu">PXDataset</span><span class="op">(</span><span class="va">px_id</span><span class="op">)</span></span></code></pre>
</div>
<p>We can then perform several functions to extract the contents of the
<code>px</code> object:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print the dataset title</span></span>
<span><span class="fu">pxtitle</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Application of SWATH Mass Spectrometry and Machine Learning in Diagnosis of Inflammatory Bowel Disease Based on Stool Proteome"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print a link to the dataset on PRIDE</span></span>
<span><span class="fu">pxurl</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print the taxonomy (organism) information for the dataset</span></span>
<span><span class="fu">pxtax</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Homo sapiens (human)"</code></pre>
</div>
<div class="section level3">
<h3 id="exploring-dataset-files">Exploring dataset files<a class="anchor" aria-label="anchor" href="#exploring-dataset-files"></a>
</h3>
<p>Let’s see which files are included in this dataset.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Retrieve a list of dataset files</span></span>
<span><span class="va">px_files</span> <span class="op">&lt;-</span> <span class="fu">pxfiles</span><span class="op">(</span><span class="va">px</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display the first few files</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">px_files</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "20201016_UdSjfb_20190115_freshstool_003.mzML"
[2] "20201016_UdSjfb_20190115_freshstool_003.wiff"
[3] "20201016_UdSjfb_20190115_freshstool_003.wiff.scan"
[4] "20201016_UdSjfb_20190115_freshstool_005.mzML"
[5] "20201016_UdSjfb_20190115_freshstool_005.wiff"
[6] "20201016_UdSjfb_20190115_freshstool_005.wiff.scan"</code></pre>
</div>
<p>PRIDE datasets often include a mix of raw data, search results,
processed outputs, and metadata.</p>
<p>To understand the data composition, we’ll examine file
extensions.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of files by extension</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="fu">sapply</span><span class="op">(</span><span class="fu">strsplit</span><span class="op">(</span><span class="va">px_files</span>, <span class="st">"\\."</span><span class="op">)</span>, <span class="va">tail</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
    fas    mzML     pdf    scan speclib     tsv     txt    wiff    xlsx
      1      78       1      78       2      13       1      78       3 </code></pre>
</div>
<p>Alternatively, we can get the same result with the tools package:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tools</span><span class="op">)</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="fu">file_ext</span><span class="op">(</span><span class="va">px_files</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
    fas    mzML     pdf    scan speclib     tsv     txt    wiff    xlsx
      1      78       1      78       2      13       1      78       3 </code></pre>
</div>
<p>Each file extension represents a specific proteomics data type. For
example, <code>.raw</code> may indicate vendor instrument output and
<code>.mzML</code> indicates an open format for MS data.</p>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Learn more about proteomics file formats </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler2" aria-labelledby="headingSpoiler2">
<div class="accordion-body">
<p>For more information about different proteomics file formats, check
out:</p>
<p><a href="https://www.ebi.ac.uk/pride/markdownpage/pridefileformats" class="external-link">PRIDE
File Formats Documentation</a></p>
<p><a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3518119/" class="external-link">Paper on
mass spec file formats (PMC3518119)</a></p>
</div>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Try visiting the dataset page on the PRIDE website and compare:</p>
<ul>
<li><p>The number of files listed there versus those returned by
rpx</p></li>
<li><p>The types of files (e.g., .raw, .mzML, .txt, etc.)</p></li>
</ul>
<p>Questions to consider:</p>
<ul>
<li><p>Why might the PRIDE web interface and R output differ?</p></li>
<li><p>What are the advantages of using programmatic access via
rpx?</p></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Programmatic access allows you to automate each step of your
analysis, making your workflow fully reproducible.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="downloading-required-files">Downloading required files<a class="anchor" aria-label="anchor" href="#downloading-required-files"></a>
</h3>
<p>Good news! We have already downloaded and processed the data for you,
as described in the next lesson. You should have downloaded this from
the Setup page.</p>
<p>If you wish to download files from another dataset, you can use the
<code>curl</code> package as demonstrated below.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">curl</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">curl</span><span class="fu">::</span><span class="fu">curl_download</span><span class="op">(</span>url<span class="op">=</span><span class="st">'https://ftp.pride.ebi.ac.uk/pride/data/archive/2024/02/PXD047585/SampleAnnotation.xlsx'</span>,destfile <span class="op">=</span> <span class="st">'data/SampleAnnotation.xlsx'</span><span class="op">)</span></span></code></pre>
</div>
<p>Note:</p>
<ul>
<li>
<p>We retrieved the <code>url</code> in an earlier step using
<code>pxurl()</code>.</p>
<ul>
<li>We have replaced <code>ftp://</code> at the start with
<code>https://</code>
</li>
<li>We have specified the name of the file we wish to download
(<code>SampleAnnotation.xlsx</code>), from the list we sourced earlier
using <code>pxfiles()</code>
</li>
</ul>
</li>
<li><p>We set <code>destfile</code> to our desired name and location for
the downloaded file.</p></li>
</ul>
<p><br></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>The dataset used in this tutorial includes stool samples from a
study investigating non-invasive diagnostic methods for Inflammatory
Bowel Disease.</li>
<li>We can use the <code>rpx</code> package to download and explore data
from PRIDE or the ProteomeXchange Consortium without leaving R.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-02diann"><p>Content from <a href="02diann.html">Processing data with DIA-NN</a></p>
<hr>
<p>Last updated on 2025-11-14 |

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/episodes/02diann.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I process my output files from a mass spectrometry-based
proteomics experiment?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to process proteomics data using DIA-NN.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="processing-proteomics-data">Processing proteomics data<a class="anchor" aria-label="anchor" href="#processing-proteomics-data"></a>
</h2>
<hr class="half-width">
<p>There are many software options for processing mass
spectrometry-based proteomics output. These software match mass spectra
against an empirical or predicted spectral library to <strong>identify
peptides</strong> and <strong>infer the protein</strong> to which they
belong.</p>
<p>Some software you may have heard of include Spectronaut, Fragpipe,
MaxQuant and DIA-NN. Each software have slightly different functionality
and attributes, and are frequently updated. In this tutorial, we are
demonstrating the use of <strong>DIA-NN</strong>, a software designed to
process DIA proteomics data.</p>
<figure><img src="fig/02diann_peptideMatching.png" alt="Proteomics software like DIA-NN match observed spectra against a spectral library to identify peptides and assign them to a protein or protein group. Source: Galaxy Training Network" class="figure mx-auto d-block"><div class="figcaption">Proteomics software like DIA-NN match observed
spectra against a spectral library to identify peptides and assign them
to a protein or protein group. Source: <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/introduction/slides.html#15" class="external-link">Galaxy
Training Network</a>
</div>
</figure><p><br></p>
<div class="section level3">
<h3 id="why-dia-nn">Why DIA-NN?<a class="anchor" aria-label="anchor" href="#why-dia-nn"></a>
</h3>
<p>Key features of DIA-NN software include:</p>
<ul>
<li>
<strong>Neural network–based scoring:</strong> Improves
identification confidence.</li>
<li>
<strong>Interference correction:</strong> Enhances quantification
accuracy.</li>
<li>
<strong>Library-free analysis:</strong> Can work without a spectral
library (generating one directly from DIA data).</li>
<li>
<strong>High speed and efficiency:</strong> Suitable for large-scale
datasets.</li>
<li>
<strong>Cross-run normalisation:</strong> Ensures consistent
quantification across experiments.</li>
<li>
<strong>Command-line executable:</strong> DIA-NN can be run from a
graphical user interface (GUI) or directly from the command line, making
it easy to integrate into an automated pipeline.</li>
</ul>
<p>DIA-NN is free to download. Detailed operating instructions and a
link to install the latest version of DIA-NN can be found on the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">GitHub
page</a>.</p>
</div>
</section><section><h2 class="section-heading" id="dia-nn-workflow">DIA-NN workflow<a class="anchor" aria-label="anchor" href="#dia-nn-workflow"></a>
</h2>
<hr class="half-width">
<div id="important-note" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="important-note" class="callout-inner">
<h3 class="callout-title">Important Note</h3>
<div class="callout-content">
<p>Due to the amount of time it will take to process the data, you do
not need to install or operate DIA-NN for the purposes of this tutorial.
Pre-processed data is provided, which has been prepared following the
procedures described below.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="part-1-creating-a-spectral-library">Part 1: Creating a spectral library<a class="anchor" aria-label="anchor" href="#part-1-creating-a-spectral-library"></a>
</h3>
<p>DIA-NN analysis consists of two steps. The first step is to
<strong>generate a predicted spectral library</strong> for your
experimental organism.</p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Download a sequence database from UniProt.</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Go to <a href="https://www.uniprot.org/uniprotkb?query=" class="external-link uri">https://www.uniprot.org/uniprotkb?query=</a>*</p></li>
<li><p>Select the <code>relevant organism</code> for your sample and
<code>Reviewed (Swiss-Prot)</code> status to filter results.</p></li>
<li><p>Click <code>Download</code></p></li>
<li><p>Change the format to
<code>FASTA (canonical and isoform)</code></p></li>
<li><p>Select <code>No</code> under Compressed and download the FASTA
file</p></li>
</ol>
</li>
</ol>
<div id="for-specific-searches" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="for-specific-searches" class="callout-inner">
<h3 class="callout-title">FOR SPECIFIC SEARCHES</h3>
<div class="callout-content">
<p>If you need to search for a variant protein or peptide that differs
from the standard organism proteome, you can manually edit the protein
sequences in the UniProt FASTA using a text editor.</p>
</div>
</div>
</div>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>Download a <a href="https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics/tree/main" class="external-link">contaminant
FASTA file</a></strong>. For most experiments, the Universal
Contaminants.fasta will be appropriate, but you may prefer to use one of
the sample-type specific contaminant FASTAs depending on your
experiment. <strong><em>More information about this resource and
contaminants will be discussed in the next
lesson.</em></strong></p></li>
<li><p><strong>Open the DIA-NN</strong> GUI or a command line interface.
If there are multiple versions of DIA-NN installed on your device,
ensure to update <code>DIA-NN exe</code> with the file path to the
version you wish to use.</p></li>
<li><p><strong>Add the UniProt and Contaminant FASTA files</strong> you
just downloaded.</p></li>
</ol>
<div id="caution1" class="callout caution">
<div class="callout-square">
<i class="callout-icon" data-feather="alert-triangle"></i>
</div>
<span class="callout-header">Caution</span>
<div class="callout-inner">
<div class="callout-content">
<p>At the time of writing, DIA-NN 2.2.0 cannot read or write files saved
in an external drive. Please ensure all input and output files are saved
on a local drive to avoid error.</p>
</div>
</div>
</div>
<ol start="5" style="list-style-type: decimal">
<li><p>Examples are included below for the additional <strong>settings
we recommend changing from the default</strong>. Depending on your
experiment and research goals, you may wish to change additional
settings from the default - read more in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings" class="external-link">DIA-NN
documentation</a>.</p></li>
<li><p><strong>Run</strong>. DIA-NN will output a predicted spectral
library in the <code>.speclib</code> file type.</p></li>
</ol>
<figure><img src="fig/02diann_DIANNgui_speclib.png" alt="Example of the DIA-NN GUI used to generate a predicted spectral library. Settings which have been altered from the default are highlighted in red." class="figure mx-auto d-block"><div class="figcaption">Example of the DIA-NN GUI used to generate a
predicted spectral library. Settings which have been altered from the
default are highlighted in red.</div>
</figure><div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">C:\DIA-NN\2.2.0\diann.exe</span> <span class="kw">`</span> <span class="co"># Replace with file path to DIA-NN on your device</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">--lib</span> <span class="st">""</span> <span class="kw">`</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">--out</span> <span class="st">"C:\path\to\output\report.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with desired file path to output report.parquet</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">--out-lib</span> <span class="st">"C:\path\to\output\report-lib.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with desired file path to output report-lib.parquet</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\UniprotFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path to Uniprot FASTA</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\ContaminantsFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path to contaminants FASTA</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">--threads</span> 30 <span class="kw">`</span> <span class="co"># Replace with the number of threads to use for this analysis</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">--verbose</span> 1 <span class="kw">`</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">--qvalue</span> 0.01 <span class="kw">`</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="ex">--matrices</span>  <span class="kw">`</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="ex">--gen-spec-lib</span> <span class="kw">`</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="ex">--predictor</span> <span class="kw">`</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="ex">--reannotate</span> <span class="kw">`</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="ex">--fasta-search</span> <span class="kw">`</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="ex">--min-fr-mz</span> 200 <span class="kw">`</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="ex">--max-fr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="ex">--met-excision</span> <span class="kw">`</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="ex">--min-pep-len</span> 7 <span class="kw">`</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="ex">--max-pep-len</span> 30 <span class="kw">`</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="ex">--min-pr-mz</span> 300 <span class="kw">`</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="ex">--max-pr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="ex">--min-pr-charge</span> 1 <span class="kw">`</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="ex">--max-pr-charge</span> 4 <span class="kw">`</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="ex">--cut</span> K<span class="pp">*</span>,R<span class="pp">*</span> <span class="kw">`</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="ex">--missed-cleavages</span> 1 <span class="kw">`</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="ex">--unimod4</span> <span class="kw">`</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="ex">--rt-profiling</span> </span></code></pre>
</div>
<p>This spectral library can be re-used for all analyses of samples from
the same organism. It is recommended to re-download the UniProt FASTA
and re-run this step approximately once a month to ensure your library
is up-to-date.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="part-2-analysis">Part 2: Analysis<a class="anchor" aria-label="anchor" href="#part-2-analysis"></a>
</h3>
<p>The second step is to <strong>search your raw files (mass
spectrometry output files)</strong> against the spectral library.</p>
<ol style="list-style-type: decimal">
<li><p><strong>Open the DIA-NN</strong> GUI or a command line interface.
If there are multiple versions of DIA-NN installed on your device,
ensure to update <code>DIA-NN exe</code> with the file path to the
version you wish to use.</p></li>
<li>
<p><strong>Upload your raw files</strong>.</p>
<ul>
<li>If using the GUI, under <code>Input</code> select <code>Raw</code>
then select all of the files you wish to search.</li>
<li>If using the command line, you can use <code>--f</code> multiple
times to list each file individually, or (recommended) use
<code>--dir [folder]</code> to instruct DIA-NN to process all raw files
saved in a particular folder.</li>
</ul>
</li>
<li><p><strong>Upload your spectral library</strong>.</p></li>
</ol>
<div id="empirical-vs-predicted-spectral-library" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="empirical-vs-predicted-spectral-library" class="callout-inner">
<h3 class="callout-title">Empirical vs predicted spectral library</h3>
<div class="callout-content">
<p>You may wish to generate an empirical spectral library from your
data, and use this for future analyses. An empirical library will be
smaller than a predicted library and therefore increase the speed of
analysis; however, may result in fewer identifications. We only
recommend using an empirical library if generated from high quality
data.</p>
<p>You can generate an empirical library from your samples by selecting
<strong>Generate spectral library</strong> and specifying the location
of its output.</p>
</div>
</div>
</div>
<ol start="4" style="list-style-type: decimal">
<li>
<p><strong>Adjust settings</strong> according to your experiment and
research goals. Again, examples are included below for the additional
settings we recommend changing from the default. You may wish to change
additional settings from the default - read more in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings" class="external-link">DIA-NN
documentation</a>.</p>
<ul>
<li>In this example, the <strong>protease</strong> Trypsin was used to
generate peptides during sample preparation.</li>
<li>Leaving <strong>Mass accuracy</strong> and <strong>MS1
accuracy</strong> on the default 0 means DIA-NN will optimise these
parameters automatically for the first run in the experiment and then
reuse the optimised settings for other runs. The DIA-NN documentation
recommends different settings depending on the type of mass spectrometer
used to generate your data. In this example, data were generated on a
<strong>TripleTOF 6600</strong>, so we set Mass accuracy to 20.0 and MS1
accuracy to 12.0</li>
<li>Set the number of <strong>Threads</strong> to be no more than the
number of logical cores on your computer.</li>
<li>In this experiment, we have set the <strong>maximum number of
variable modifications</strong> to 0; however, it is common for this
setting to be relaxed in the range of 1 - 3.</li>
<li>DIA-NN’s default setting is to activate
<strong>match-between-runs</strong>. From the documentation: <em>In MBR
mode, DIA-NN does two passes over the data. During the first pass, it
creates an empirical spectral library from the data. During the second
pass, it reanalyses the experiment with this empirical library, which
may result in much improved identification numbers, data completeness
and quantification accuracy.</em>
</li>
</ul>
</li>
</ol>
<figure><img src="fig/02diann_DIANNgui_analysis.png" alt="Example of the DIA-NN GUI used to analyse the data described in this tutorial. Settings which have been altered from the default are highlighted in red." class="figure mx-auto d-block"><div class="figcaption">Example of the DIA-NN GUI used to analyse the
data described in this tutorial. Settings which have been altered from
the default are highlighted in red.</div>
</figure><div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">C:\DIA-NN\2.2.0\diann.exe</span> <span class="kw">`</span> <span class="co"># Replace with file path to DIA-NN on your device</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">--dir</span> <span class="st">"C:\path\to\folder\of\raw\files"</span> <span class="kw">`</span> <span class="co"># Replace with file path to folder containing all raw files to be analysed</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">--lib</span> <span class="st">"C:\path\to\SpectralLibrary\report-lib.predicted.speclib"</span> <span class="kw">`</span> <span class="co"># Replace with file path to spectral library</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="ex">--out</span> <span class="st">"C:\path\to\output\report.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with desired file path to output report.parquet</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="ex">--threads</span> 30 <span class="kw">`</span> <span class="co"># Replace with the number of threads to use for this analysis</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="ex">--verbose</span> 1 <span class="kw">`</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="ex">--qvalue</span> 0.01 <span class="kw">`</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="ex">--matrices</span>  <span class="kw">`</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">--unimod4</span> <span class="kw">`</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="ex">--mass-acc</span> 20 <span class="kw">`</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="ex">--mass-acc-ms1</span> 12.0 <span class="kw">`</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="ex">--reanalyse</span> <span class="kw">`</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="ex">--rt-profiling</span> </span></code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>If you want to run DIA-NN from the command line but are unsure how to
write the code, you can update your settings in the GUI, click ‘Run’,
and copy the commands output in the log. That’s how we got the example
code above!</p>
<p>Can you match each line of code to the corresponding setting in the
GUI?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Check the description of available commands in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#command-line-reference" class="external-link">DIA-NN
documentation</a></p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="dia-nn-output">DIA-NN output<a class="anchor" aria-label="anchor" href="#dia-nn-output"></a>
</h2>
<hr class="half-width">
<p>DIA-NN output includes several files, including:</p>
<table class="table">
<colgroup>
<col width="21%">
<col width="78%">
</colgroup>
<thead><tr class="header">
<th>Output file</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>log.txt</strong></td>
<td>A log of what DIA-NN has run. You should always use
<code>ctrl + F</code> to check for any ‘ERROR’ or ‘WARNING’ messages
issued by DIA-NN during your run.</td>
</tr>
<tr class="even">
<td><strong>.parquet</strong></td>
<td>This is what we will be using for subsequent analyses in this
tutorial.</td>
</tr>
<tr class="odd">
<td><strong>pg_matrix.tsv</strong></td>
<td>Matrix of protein values generated by DIA-NN.</td>
</tr>
<tr class="even">
<td><strong>pr_matrix.tsv</strong></td>
<td>Matrix of peptide values generated by DIA-NN.</td>
</tr>
<tr class="odd">
<td><strong>trends.pdf</strong></td>
<td>A PDF report listing the number of peptides, proteins, and other
quality control measures detected for each sample.</td>
</tr>
</tbody>
</table>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>If DIA-NN is run through the command line rather than the GUI, the
<em>trends.pdf</em> and <em>runs.pdf</em> reports are not automatically
generated. Using the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">DIA-NN
documentation</a>, can you work out how to generate this report via the
command line?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Run the <code>diann-stats.py</code> Python script provided when you
download DIA-NN.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">python</span> C:<span class="dt">\D</span>IA-NN<span class="dt">\2</span>.2.0<span class="dt">\d</span>iann-stats.py C:<span class="dt">\p</span>ath<span class="dt">\t</span>o<span class="dt">\r</span>eport.parquet</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>We have now processed our raw files, identifying the peptides present
in our samples and inferring the proteins to which they belong.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Learn more about DIANN </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p>For more background information, see the original publication:<br><em>Demichev, V., et al. (2020). DIA-NN: neural networks and
interference correction enable deep proteome coverage in high
throughput. Nature Methods.</em> <a href="https://doi.org/10.1038/s41592-019-0638-x" class="external-link">https://doi.org/10.1038/s41592-019-0638-x</a>.</p>
<p>For further details about the different DIA-NN settings, see the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">documentation</a>.</p>
<p>If you come across any issues, the creators are fairly active at
responding to <a href="https://github.com/vdemichev/DiaNN/issues" class="external-link">issues</a> and <a href="https://github.com/vdemichev/DiaNN/discussions" class="external-link">discussions</a> on
their GitHub.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>Raw files from a mass spectrometry-based DIA proteomics experiment
can be processed using DIA-NN (peptide identification and protein
inference).</li>
<li>The DIA-NN workflow has two key steps: generating a predicted
spectral library, and processing raw files.</li>
<li>The <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">DIA-NN
documentation</a> contains detailed information about changing default
settings as appropriate for your experiment.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-03datacleaning"><p>Content from <a href="03datacleaning.html">Data cleaning with R</a></p>
<hr>
<p>Last updated on 2025-11-14 |

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/episodes/03datacleaning.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do we convert our processed proteomics data (DIA-NN output) into
an analysis-ready format?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the steps required to clean proteomics data and generate
analysis-ready protein values using <code>limpa</code>.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="load-processed-data-and-sample-annotation">Load processed data and sample annotation<a class="anchor" aria-label="anchor" href="#load-processed-data-and-sample-annotation"></a>
</h2>
<hr class="half-width">
<p>After processing raw proteomics data, further cleaning and analysis
can be conducted using <strong>R</strong>, <strong>Python</strong>, or
by importing data to other platforms like <strong>Perseus</strong> or
<strong>Skyline</strong>.</p>
<p>In this tutorial, we will be teaching you one method of cleaning and
analysing proteomics data using R. <strong>To follow along the remainder
of this tutorial, you should open a new R script in RStudio and run the
example code provided.</strong></p>
<div id="the-limpa-package" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="the-limpa-package" class="callout-inner">
<h3 class="callout-title">The limpa package</h3>
<div class="callout-content">
<p>In this tutorial, we are using the <code>limpa</code> package to load
and clean our data. This package was published in 2025 by the <a href="https://github.com/SmythLab/limpa" class="external-link">Smyth Lab</a>, based at the
Walter and Eliza Hall Institute of Medical Research (WEHI).</p>
<p>For background information, see the original publications:</p>
<p><em>Li, M., Cobbold, S. A., &amp; Smyth, G. K. (2025). Quantification
and differential analysis of mass spectrometry proteomics data with
probabilistic recovery of information from missing values. bioRxiv,
2025.2004.2028.651125. <a href="https://doi.org/10.1101/2025.04.28.651125" class="external-link uri">https://doi.org/10.1101/2025.04.28.651125</a> </em></p>
<p><em>Li, M., &amp; Smyth, G. K. (2023). Neither random nor censored:
estimating intensity-dependent probabilities for missing values in
label-free proteomics. Bioinformatics, 39(5). <a href="https://doi.org/10.1093/bioinformatics/btad200" class="external-link uri">https://doi.org/10.1093/bioinformatics/btad200</a> </em></p>
<p>You can also find further information in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">vignette</a>
or <a href="https://bioconductor.org/packages/release/bioc/manuals/limpa/man/limpa.pdf" class="external-link">reference
manual</a> accessed via the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">Bioconductor
page</a>.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First, load the required R packages.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">limpa</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">EnhancedVolcano</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">STRINGdb</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">clusterProfiler</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">arrow</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">rpx</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">stringr</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-dia-nn-output">Load DIA-NN output<a class="anchor" aria-label="anchor" href="#load-dia-nn-output"></a>
</h3>
<p><strong>Load the peptide-level data</strong> processed by DIA-NN
(<code>.parquet</code> file), <strong>filtering observations</strong> by
quality metrics as recommended in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">limpa
vignette</a>.</p>
<p>The peptide values will automatically be <strong>converted to
log2</strong> upon import.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">readDIANN</span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/MBIntroToProteomics.parquet'</span>,format<span class="op">=</span><span class="st">"parquet"</span>,</span>
<span>                       q.columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Q.Value"</span>,<span class="st">"Lib.Q.Value"</span>,<span class="st">"Lib.PG.Q.Value"</span><span class="op">)</span>,</span>
<span>                       q.cutoffs <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Length of q-value columns does not match with length of q-value cutoffs. Use q.cutoffs[1] for all columns.</code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>These quality filtering metrics are recommended in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">limpa
vignette</a> for data searched in DIA-NN <strong>with
match-between-runs</strong>.</p>
<p>What <code>q.columns</code> are recommended for data searched
<strong>without MBR</strong>?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">readDIANN</span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/MBIntroToProteomics.parquet'</span>,format<span class="op">=</span><span class="st">"parquet"</span>,</span>
<span>                       q.columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Q.Value"</span>,<span class="st">"Global.Q.Value"</span>,<span class="st">"Global.PG.Q.Value"</span><span class="op">)</span>,</span>
<span>                       q.cutoffs <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Let’s look at the structure of the data.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "E"     "genes"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see, our data has been imported as a limma EList object,
with components <strong>‘E’ (log2 peptide matrix)</strong> and
<strong>‘genes’ (feature annotation)</strong> containing information
about each peptide.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="load-sample-annotation">Load sample annotation<a class="anchor" aria-label="anchor" href="#load-sample-annotation"></a>
</h3>
<p>We also need to <strong>import information about our samples</strong>
for later analyses.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples_stool</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">readxl</span><span class="fu">::</span><span class="fu">read_excel</span><span class="op">(</span><span class="st">'data/SampleAnnotation.xlsx'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  file_order Sample_id-new Batch Class
1         33       B1_29_c    B1   CDr
2         93       B1_87_e    B1   UCr
3         71       B1_66_e    B1   UCr
4          9       B1_08_b    B1   aCD
5         26       B1_22_c    B1   CDr
6         31       B1_27_c    B1   CDr
                                                                                                                                         file_name
1 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_067-29.mzML
2    D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_187.mzML
3 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_143-66.mzML
4  D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_019-8.mzML
5 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_053-22.mzML
6 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_063-27.mzML</code></pre>
</div>
<p>We need to clean our sample annotation data to add an identifying
column which <strong>matches the sample column names in our peptide
matrix.</strong></p>
<p>Given we are only using a subset of the main dataset in this
tutorial, we can also filter the sample annotation dataframe to
<strong>only include rows pertaining to the samples in our
subset.</strong></p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create sample_name column that matches peptide matrix column names</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">"\\\\"</span>, <span class="st">"/"</span>, <span class="va">samples_stool</span><span class="op">$</span><span class="va">file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">basename</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">".mzML"</span>, <span class="st">""</span>,<span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to only samples present in our dataset</span></span>
<span><span class="va">samples_stool</span> <span class="op">&lt;-</span> <span class="va">samples_stool</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">%in%</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>We can attach our sample annotation data to the EList object for
later analyses.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure sample row order matches column order of the peptide matrix</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">samples_stool</span><span class="op">[</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">%in%</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">sample_name</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span>,<span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Add experimental metadata to EList</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'Class'</span>, <span class="st">'Batch'</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="quality-filtering">Quality filtering<a class="anchor" aria-label="anchor" href="#quality-filtering"></a>
</h2>
<hr class="half-width">
<p>After reading in the data, it is common to conduct additional
filtering to improve the quality of your data and assist with downstream
interpretation of results.</p>
<ul>
<li>
<strong>Filtering out non-proteotypic peptides:</strong> Removes
peptides that belong to more than one protein.</li>
<li>
<strong>Filtering out compound proteins:</strong> Removes peptides
belonging to compound protein groups consisting of multiple proteins
separated by “;” delimiters.</li>
<li>
<strong>Filtering out singleton peptides:</strong> Removes peptides
that are the only peptide belonging to that protein (i.e. keeps only
proteins with multiple peptides).</li>
</ul>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterNonProteotypicPeptides</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterCompoundProteins</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterSingletonPeptides</span><span class="op">(</span><span class="va">y.peptide</span>, min.n.peptides <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<p>These filtering steps are optional; if they are not run,
<code>limpa</code> will still be able to conduct protein quantification.
For more information about these peptide-level filters and whether they
are appropriate for your experiment, please consult the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">limpa
documentation</a>.</p>
</section><section><h2 class="section-heading" id="removing-contaminants">Removing contaminants<a class="anchor" aria-label="anchor" href="#removing-contaminants"></a>
</h2>
<hr class="half-width">
<p>Contaminant background signals can influence mass spectrometry-based
proteomics data. It is essential to <strong>remove contaminant
proteins</strong> from the sample preparation process to avoid
confounding results.</p>
<p><a href="https://doi.org/10.1021/acs.jproteome.2c00145" class="external-link">Frankenfield
et al. (2022)</a> created a new set of sample-type specific and
universal contaminant FASTA files which they showed reduced false
identifications, increased protein identifications and did not influence
protein quantification for DIA workflows. These FASTA files updated upon
widely used contaminant protein lists from MaxQuant and the common
Repository of Adventitious Proteins (cRAP).</p>
<p>In the previous lesson, we incorporated the universal contaminants
FASTA into our spectral library. This FASTA is structured such that all
contaminant proteins are named beginning with <strong>‘Cont_’</strong>,
which means we can simply remove any proteins from our analysis
containing this string text in their name.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove contaminant proteins from the genes dataframe</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">Protein.Group</span>, <span class="st">"Cont_"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Match peptides in the matrix to only those remaining in the dataframe</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span>, <span class="op">]</span></span></code></pre>
</div>
<div id="more-about-contaminants" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="more-about-contaminants" class="callout-inner">
<h3 class="callout-title">More about contaminants</h3>
<div class="callout-content">
<p>You can <strong>download the contaminant FASTA files from <a href="https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics" class="external-link">GitHub</a>.</strong></p>
<p>If you are analysing data that has been processed without a
contaminant FASTA, you can also download an <strong>Excel spreadsheet
contaminant list</strong> from the same link and use this to identify
and remove contaminant proteins from your dataset.</p>
<p>For more information, see the original publication:</p>
<p><em>Frankenfield, A. M., Ni, J., Ahmed, M., &amp; Hao, L. (2022).
Protein Contaminants Matter: Building Universal Protein Contaminant
Libraries for DDA and DIA Proteomics. Journal of proteome research,
21(9), 2104–2113. <a href="https://doi.org/10.1021/acs.jproteome.2c00145" class="external-link uri">https://doi.org/10.1021/acs.jproteome.2c00145</a></em></p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="protein-quantification">Protein quantification<a class="anchor" aria-label="anchor" href="#protein-quantification"></a>
</h2>
<hr class="half-width">
<p>Missing data has been a longstanding issue impacting the accuracy and
reproducibility of mass spectrometry-based proteomic analyses,
complicating downstream analyses. Most established imputation methods
are not designed to account for the specific attributes of proteomics
data, and there is no consensus in the literature for how missing values
should be managed in proteomics experiments. This presents an issue with
using the protein values directly generated by platforms like DIA-NN,
which contain many missing values.</p>
<p>In this tutorial, we are using the <code>dpcQuant()</code> method of
protein quantification from the <code>limpa</code> package, or
<em>Linear Models for Proteomics Data</em> <a href="https://doi.org/10.1101/2025.04.28.651125" class="external-link">(Li, Cobbold &amp;
Smyth, 2025)</a>. The <strong>detection probability curve (DPC)</strong>
models the relationship between peptide intensity and missingness,
showing that almost all missing values in label-free proteomics data are
not missing at random and should be accounted for <a href="https://doi.org/10.1093/bioinformatics/btad200" class="external-link">(Li &amp; Smyth,
2023)</a>. The <code>dpcQuant()</code> uses Bayesian statistical methods
based on peptide intensities and the DPC coefficients for a particular
dataset to generate a complete matrix of protein values without the need
for imputation. This method outputs a linear model object suitable for
downstream analysis with the <code>limma</code> package.</p>
<p>To learn more about the detection probability curve and the
<code>limpa</code> package, please read the original publications.</p>
<div class="section level3">
<h3 id="using-limpa-and-the-dpc-for-protein-quantification">Using limpa and the DPC for protein quantification<a class="anchor" aria-label="anchor" href="#using-limpa-and-the-dpc-for-protein-quantification"></a>
</h3>
<p>First, we will generate and visualise our <strong>detection
probability curve</strong>.</p>
<p>The DPC coefficients are related to the dataset and software used for
processing. A steeper slope means there is more statistical information
in the pattern of missing data. A slope closer to 0 means there is more
data missing at random.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate the DPC for our data</span></span>
<span><span class="va">dpcfit</span> <span class="op">&lt;-</span> <span class="fu">dpcON</span><span class="op">(</span><span class="va">y.peptide</span>, robust<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the intercept and slope of our DPC</span></span>
<span><span class="va">dpcfit</span><span class="op">$</span><span class="va">dpc</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>     beta0      beta1
-7.2678650  0.6033267 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the DPC</span></span>
<span><span class="fu">plotDPC</span><span class="op">(</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/03datacleaning-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Next, we can use the DPC to generate our log2 protein values. This
function requires the following input:</p>
<ul>
<li>a limma EList object, or a numeric matrix, of peptide-level log2
expression values, with samples as columns and peptides as rows
(<code>y.peptide</code>)</li>
<li>vector of protein IDs (<code>"Protein.Names"</code>)</li>
<li>numeric vector providing the slope and intercept of the DPC
(<code>dpcfit</code>)</li>
</ul>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.protein</span> <span class="op">&lt;-</span> <span class="fu">dpcQuant</span><span class="op">(</span><span class="va">y.peptide</span>, <span class="st">"Protein.Names"</span>, dpc<span class="op">=</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Estimating hyperparameters ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Quantifying proteins ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Proteins: 280 Peptides: 2672</code></pre>
</div>
<p><em>Note: our dataset is quite small, so this process should only
take a few seconds. For larger datasets, it may take minutes or hours
for this command to run.</em></p>
<div id="alternate-versions-of-the-dpc" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="alternate-versions-of-the-dpc" class="callout-inner">
<h3 class="callout-title">Alternate versions of the DPC</h3>
<div class="callout-content">
<p>The base <code>dpc()</code> assumes observed log-intensities are
normally distributed.</p>
<p>A modified version, used in this tutorial,
<code>dpcON(robust=TRUE)</code> downweighs outliers.</p>
<p>Alternatively, <code>dpcCN()</code> assumes the complete set of
log-intensities are normally distributed. Using this method may
underestimate the DPC slope if there is a non-small proportion of
peptides missing by chance.</p>
<p>You can also choose your own slope by passing the argument
<code>dpc.slope=</code> to <code>dpcQuant()</code> instead of
<code>dpc=dpcfit</code>. Using this method, an intercept value will
automatically be generated. You can check this value by using
<code>estimateDPCIntercept()</code>. The authors recommend a slope of
0.6 - 0.8 is usually appropriate for data generated using DIA-NN.</p>
<p>You may wish to conduct further statistical analyses to determine
which method is most appropriate for your experiment; however, the
authors suggest that downstream analysis should not be significantly
affected by differing DPC coefficients.</p>
<p>You can read more about the differences between each version of the
DPC in the <a href="https://www.biorxiv.org/content/10.1101/2025.04.28.651125v1" class="external-link">paper</a>
or the <a href="https://bioconductor.org/packages/release/bioc/manuals/limpa/man/limpa.pdf" class="external-link">limpa
manual</a>.</p>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>The limpa package also provides a function to impute missing peptide
values and generate a complete peptide matrix using the DPC. Can you
search the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">limpa
documentation</a> to identify the command used to impute and quantify
missing values in a rowwise manner?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p><code>dpcImpute()</code></p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="dpcquant-output">dpcQuant() output<a class="anchor" aria-label="anchor" href="#dpcquant-output"></a>
</h3>
<p>We have now generated an EList object for our protein values. Let’s
investigate the other components of this object:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "E"       "genes"   "other"   "targets" "dpc"    </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">$</span><span class="va">other</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "n.observations" "standard.error"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Protein.Group" "Protein.Names" "Genes"         "NPeptides"
[5] "PropObs"      </code></pre>
</div>
<table class="table">
<colgroup>
<col width="28%">
<col width="72%">
</colgroup>
<thead><tr class="header">
<th align="center">EList component</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center"><strong>E</strong></td>
<td>A complete matrix of log2 protein values with no missing data, with
samples as columns and proteins as rows</td>
</tr>
<tr class="even">
<td align="center"><strong>genes</strong></td>
<td>Feature annotation dataframe, with columns for: UniProt IDs
(Protein.Group), proteins (Protein.Names), gene names (Genes), number of
peptides used to quantify each protein (NPeptides), and the proportion
of possible observations used to generate each protein value
(PropObs)</td>
</tr>
<tr class="odd">
<td align="center"><strong>other</strong></td>
<td>List of two matrices with columns and rows corresponding to protein
matrix E: number of peptides observed for each protein in each sample
(n.observations), and the standard error associated with each protein
value for each sample (standard.error)</td>
</tr>
<tr class="even">
<td align="center"><strong>targets</strong></td>
<td>Sample annotation dataframe</td>
</tr>
<tr class="odd">
<td align="center"><strong>dpc</strong></td>
<td>DPC intercept and slope coefficient used for protein
quantification</td>
</tr>
</tbody>
</table>
<div id="challenge3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Compare the first few rows of the protein (<code>E</code>),
<code>n.observations</code> and <code>standard.error</code> matrices.
Take a look at the <code>NPeptides</code> and <code>PropObs</code>
values for those proteins also. What do you notice about the
relationship between these quality metrics?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Proteins with smaller <code>NPeptides</code> and <code>PropObs</code>
values, and smaller <code>n.observations</code> for individual samples,
have higher standard error.</p>
<p>The more peptide data available for a particular protein in a
particular sample, the greater the confidence in the protein value, and
therefore the lower the standard error.</p>
<p>If you are interested in a particular protein, it is important to
look at these quality metrics for each of your samples.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="dimensionality-reduction-and-qc">Dimensionality Reduction and QC<a class="anchor" aria-label="anchor" href="#dimensionality-reduction-and-qc"></a>
</h2>
<hr class="half-width">
<p>Let’s take a look at our dataset. The <code>plotMDSUsingSEs()</code>
function, or <em>Multidimensional Scaling Plot of Gene Expression
Profiles, Using Standard Errors</em>, is similar to
<code>plotMDS()</code> in the <code>limma</code> package, but takes
account of the standard errors generated by <code>dpcQuant()</code>.
This is essentially a PCA plot for the newly generated protein values
that takes into consideration their standard error.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/03datacleaning-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>That plot is extremely messy and doesn’t tell us much! Let’s clean it
up.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Factorise and set colors for class and batch</span></span>
<span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Class</span><span class="op">)</span></span>
<span><span class="va">Class.color</span> <span class="op">&lt;-</span> <span class="va">Class</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">hcl.colors</span><span class="op">(</span><span class="fu">nlevels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, palette <span class="op">=</span> <span class="st">"cividis"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">Batch.color</span> <span class="op">&lt;-</span> <span class="va">Batch</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">hcl.colors</span><span class="op">(</span><span class="fu">nlevels</span><span class="op">(</span><span class="va">Batch</span><span class="op">)</span>, palette <span class="op">=</span> <span class="st">"Set 2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Class visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/03datacleaning-rendered-unnamed-chunk-15-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Batch visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/03datacleaning-rendered-unnamed-chunk-15-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Class (labels) and batch (colors) visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, labels<span class="op">=</span><span class="va">Class</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/03datacleaning-rendered-unnamed-chunk-15-3.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can already see a distinction between the aCD and Ctrl groups,
which will be explored further in the next lesson.</p>
<p>It also looks like there are some differences between the batches.
<strong>We should add batch as a covariate to our analyses in the next
lesson.</strong></p>
<div id="managing-batch-effects" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="managing-batch-effects" class="callout-inner">
<h3 class="callout-title">Managing batch effects</h3>
<div class="callout-content">
<p>Explaining the different methods of identifying, accounting and
correcting for batch effects in proteomics data is an entire tutorial
itself! For further information, we recommend the <a href="https://github.com/melbintgen/Batch-effect-management?tab=readme-ov-file" class="external-link">Melbourne
Integrative Genomics Workshop: Managing batch effects in biological
studies</a>.</p>
<p>If you are concerned about batch effects in your data, University of
Melbourne staff and students or affiliates can also <a href="https://mdhs.unimelb.edu.au/melbournebioinformatics/training-and-support/expert-advice" class="external-link">contact
us</a> for a free consultation.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="normalisation">Normalisation<a class="anchor" aria-label="anchor" href="#normalisation"></a>
</h3>
<p>Normalisation is another key element of the proteomics data cleaning
workflow. Normalisation techniques aim to remove non-biological
variation in the data, for example, variation due to sample loading or
batch effects.</p>
<p>Log transformation (which was automatically applied when we imported
our peptide data) helps to normalise intensity values. The default
RT-dependent normalisation applied during DIA-NN processing is also
quite effective and often sufficient for DIA data; however, each
experimental environment and techniques will have a different optimal
approach. You may wish to apply an additional method of normalisation to
your data.</p>
<p>To learn more about normalisation, keep an eye on the <a href="https://www.eventbrite.com.au/o/melbourne-bioinformatics-13058846490" class="external-link">Melbourne
Bioinformatics Eventbrite page</a> for a normalisation workshop coming
in 2026.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> More about normalisation </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>To learn more about different normalisation techniques for proteomics
data, we recommend the following papers:</p>
<p><em>Peng, H., Wang, H., Kong, W., Li, J., &amp; Goh, W. W. B. (2024).
Optimizing differential expression analysis for proteomics data via
high-performing rules and ensemble inference. Nature communications,
15(1), 3922. <a href="https://doi.org/10.1038/s41467-024-47899-w" class="external-link uri">https://doi.org/10.1038/s41467-024-47899-w</a></em></p>
<p><em>Tseng, C. Y., Salguero, J. A., Breidenbach, J. D., Solomon, E.,
Sanders, C. K., Harvey, T., Thornhill, M. G., Palmisano, S. J., Sasiene,
Z. J., Blackwell, B. R., McBride, E. M., Luchini, K. A., LeBrun, E. S.,
Alvarez, M., Mach, P. M., Rivera, E. S., &amp; Glaros, T. G. (2025).
Evaluation of normalization strategies for mass spectrometry-based
multi-omics datasets. Metabolomics : Official journal of the Metabolomic
Society, 21(4), 98. <a href="https://doi.org/10.1007/s11306-025-02297-1" class="external-link uri">https://doi.org/10.1007/s11306-025-02297-1</a></em></p>
<p><em>Välikangas, T., Suomi, T., &amp; Elo, L. L. (2018). A systematic
evaluation of normalization methods in quantitative label-free
proteomics. Briefings in bioinformatics, 19(1), 1–11. <a href="https://doi.org/10.1093/bib/bbw095" class="external-link uri">https://doi.org/10.1093/bib/bbw095</a></em></p>
</div>
</div>
</div>
</div>
<p>Good work! We now have a clean dataset ready for analysis.</p>
<p><br></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>Key elements of the proteomics data cleaning workflow include:
quality filtering, removing contaminants, protein quantification,
managing batch effects, and normalisation.</li>
<li>We can use the <code>limpa</code> package in R to clean our peptide
data and generate a complete, analysis-ready protein matrix with
corresponding standard error values.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-04DEA"><p>Content from <a href="04DEA.html">Differential expression analysis</a></p>
<hr>
<p>Last updated on 2025-11-14 |

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/episodes/04DEA.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can we identify the differentially expressed proteins between
our experimental groups?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Conduct differential expression analysis using
<code>limpa</code>
</li>
<li>Visualise the top differentially expressed proteins</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="differential-expression-analysis-dea">Differential Expression Analysis (DEA)<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-dea"></a>
</h2>
<hr class="half-width">
<p>Now it is time to analyse our data and investigate proteins that are
differentially expressed between our two experimental groups (control vs
active Crohn’s disease). We will make use of the <code>limma</code>
package and closely related <code>limpa</code> functions specifically
designed for proteomics data.</p>
<p>First, we will <strong>fit a linear model</strong> that includes both
Class and Batch effects.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set the control group as our reference</span></span>
<span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">Class</span>, ref <span class="op">=</span> <span class="st">"Ctrl"</span><span class="op">)</span></span>
<span><span class="co">## We also need to reset our colors to match the correct labels</span></span>
<span><span class="va">Class.color</span> <span class="op">&lt;-</span> <span class="va">Class</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">hcl.colors</span><span class="op">(</span><span class="fu">nlevels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, palette <span class="op">=</span> <span class="st">"cividis"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Fit a linear model</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">Class</span> <span class="op">+</span> <span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">design</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   (Intercept) ClassaCD BatchB3
1            1        1       0
2            1        1       0
3            1        0       0
4            1        1       0
5            1        0       0
6            1        0       0
7            1        1       0
8            1        1       0
9            1        0       0
10           1        0       0
11           1        0       0
12           1        1       1
13           1        0       1
14           1        1       1
15           1        1       1
16           1        0       1
17           1        0       1
18           1        1       1
19           1        0       1
20           1        0       1
attr(,"assign")
[1] 0 1 2
attr(,"contrasts")
attr(,"contrasts")$Class
[1] "contr.treatment"

attr(,"contrasts")$Batch
[1] "contr.treatment"</code></pre>
</div>
<p>We can see both Class and Batch accounted for in our design
matrix.</p>
<p>Now we run the <code>limpa</code> <strong>differential expression
analysis</strong> function <code>dpcDE()</code> to identify the top
differentially expressed proteins in our dataset.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">dpcDE</span><span class="op">(</span><span class="va">y.protein</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">eBayes</span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save our DEA results as a dataframe</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">2</span>, number <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">results</span>, n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>            Protein.Group Protein.Names   Genes NPeptides   PropObs    logFC
PRTN3_HUMAN        P24158   PRTN3_HUMAN   PRTN3        11 0.5318182 4.466373
S10A9_HUMAN        P06702   S10A9_HUMAN  S100A9        23 0.6086957 3.809771
S10A8_HUMAN        P05109   S10A8_HUMAN  S100A8        17 0.6500000 3.811580
TRFL_HUMAN         P02788    TRFL_HUMAN     LTF        46 0.3152174 3.998157
CEAM8_HUMAN        P31997   CEAM8_HUMAN CEACAM8         6 0.3583333 3.130768
SAMP_HUMAN         P02743    SAMP_HUMAN    APCS         8 0.5500000 2.720497
A1AG1_HUMAN        P02763   A1AG1_HUMAN    ORM1        11 0.4454545 3.349033
CAP7_HUMAN         P20160    CAP7_HUMAN    AZU1         6 0.4666667 3.200933
FRIH_HUMAN         P02794    FRIH_HUMAN    FTH1        12 0.5625000 2.528228
RNAS2_HUMAN        P10153   RNAS2_HUMAN  RNASE2         6 0.5666667 2.887339
              AveExpr        t      P.Value    adj.P.Val         B
PRTN3_HUMAN 11.146376 6.909177 8.925046e-08 2.499013e-05 7.9144156
S10A9_HUMAN 12.809535 6.354519 4.259264e-07 5.962969e-05 6.4365522
S10A8_HUMAN 12.865666 6.070836 9.561924e-07 8.924462e-05 5.6692556
TRFL_HUMAN   9.061431 5.755654 2.361294e-06 1.601370e-04 4.8072839
CEAM8_HUMAN 10.078482 5.689096 2.859590e-06 1.601370e-04 4.6265751
SAMP_HUMAN  11.028982 5.317255 8.351615e-06 3.897420e-04 3.6088844
A1AG1_HUMAN 10.284907 5.228167 1.079921e-05 4.319682e-04 3.3650142
CAP7_HUMAN  10.422828 4.613155 6.335616e-05 2.217466e-03 1.6820861
FRIH_HUMAN  11.341144 4.227543 1.897414e-04 5.396242e-03 0.6493737
RNAS2_HUMAN 11.228811 4.222021 1.927229e-04 5.396242e-03 0.6366061</code></pre>
</div>
<div id="a-closer-look-at-the-results" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="a-closer-look-at-the-results" class="callout-inner">
<h3 class="callout-title">A closer look at the results</h3>
<div class="callout-content">
<p>Our output table is ordered by B statistic. Use
<code>View(results)</code></p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="visualise-differentially-expressed-proteins">Visualise differentially expressed proteins<a class="anchor" aria-label="anchor" href="#visualise-differentially-expressed-proteins"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="expression-values-and-standard-error-by-sample">Expression values and standard error by sample<a class="anchor" aria-label="anchor" href="#expression-values-and-standard-error-by-sample"></a>
</h3>
<p>We can visualise the log2 expression values for individual proteins,
together with standard errors.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotProtein</span><span class="op">(</span><span class="va">y.protein</span>, <span class="st">"PRTN3_HUMAN"</span>, col <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'topleft'</span>, legend <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotProtein</span><span class="op">(</span><span class="va">y.protein</span>, <span class="st">"S10A9_HUMAN"</span>, col <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'topleft'</span>, legend <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-4-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>This simple function plots the protein values from our expression
matrix, which we did not directly edit to mitigate batch effects. Try
running the <code>limma</code> function to remove batch effects from our
dataset and see how this changes results.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.protein.rbe</span> <span class="op">&lt;-</span> <span class="va">y.protein</span></span>
<span><span class="va">y.protein.rbe</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="fu">removeBatchEffect</span><span class="op">(</span><span class="va">y.protein</span>,batch <span class="op">=</span> <span class="va">y.protein</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Batch</span>, group<span class="op">=</span><span class="va">y.protein</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Class</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plotProtein</span><span class="op">(</span><span class="va">y.protein.rbe</span>, <span class="st">"S10A9_HUMAN"</span>, col <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'topleft'</span>, legend <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-5-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>We can filter our results to the up to 50 top most significant
differentially expressed proteins, then visualise via a clustered
heatmap.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter for the up to 50 most significant results</span></span>
<span><span class="va">sig_proteins</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># filter by p value</span></span>
<span>  <span class="fu">top_n</span><span class="op">(</span><span class="fl">50</span>, wt <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># filter by absolute log fold change</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Protein.Names</span><span class="op">)</span></span>
<span><span class="va">expr_matrix</span> <span class="op">&lt;-</span> <span class="va">y.protein</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="va">sig_proteins</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Scale the data and visualise via heatmap</span></span>
<span><span class="va">scaled_expr</span> <span class="op">&lt;-</span> <span class="fu">t</span><span class="op">(</span><span class="fu">scale</span><span class="op">(</span><span class="fu">t</span><span class="op">(</span><span class="va">expr_matrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">scaled_expr</span>,</span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         show_rownames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         show_colnames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">y.protein</span><span class="op">$</span><span class="va">targets</span>,</span>
<span>         clustering_method <span class="op">=</span> <span class="st">'ward.D'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="discussion2" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>How would you interpret this heatmap?</p>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h3>
<p>We can also visualise our results as a volcano plot using
<code>EnhancedVolcano()</code>.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">results</span>,</span>
<span>                lab <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">Genes</span>,</span>
<span>                x <span class="op">=</span> <span class="st">'logFC'</span>,</span>
<span>                y <span class="op">=</span> <span class="st">'adj.P.Val'</span>,</span>
<span>                pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                FCcutoff <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>                pointSize <span class="op">=</span> <span class="fl">2.0</span>,</span>
<span>                labSize <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>                drawConnectors <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                maxoverlapsConnectors <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>                lengthConnectors <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.001</span>, <span class="st">'npc'</span><span class="op">)</span>,</span>
<span>                typeConnectors <span class="op">=</span><span class="st">'closed'</span>,</span>
<span>                ylim <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>                xlim <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/04DEA-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>We can fit a linear model to our data and conduct differential
expression analysis using <code>limpa</code> and <code>limma</code>
functions.</li>
<li>There are many different ways to visualise differentially expressed
proteins.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-05EnrichmentAnalysis"><p>Content from <a href="05EnrichmentAnalysis.html">Network and Enrichment Analysis</a></p>
<hr>
<p>Last updated on 2025-11-14 |

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/episodes/05EnrichmentAnalysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can DEA results be used to identify biologically meaningful
patterns?</li>
<li>How can we visualise and interpret protein–protein interaction (PPI)
networks and pathway enrichment results?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Visualise significant proteins using network analysis (STRING)</li>
<li>Identify enriched biological processes and pathways (GO and
KEGG)</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Now that we have a list of differentially expressed (DE) proteins, we
can explore how these proteins interact with each other and what
biological functions or pathways they are involved in.</p>
<p>This step helps translate statistical significance into biological
meaning by integrating our results with known protein–protein
interactions (PPIs) and functional annotation databases.</p>
<section><h2 class="section-heading" id="string-proteinprotein-interaction-network-analysis">STRING protein–protein interaction network analysis<a class="anchor" aria-label="anchor" href="#string-proteinprotein-interaction-network-analysis"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://string-db.org/" class="external-link">STRING</a> database integrates
known and predicted PPIs from multiple sources (e.g. experimental data,
text mining, prediction methods, and curated databases).</p>
<p>We can visualise DE proteins within the context of their
<strong>interaction network</strong> to identify clusters or modules
potentially corresponding to biological pathways.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create an instance of the STRINGdb reference class for humans (species 9606)</span></span>
<span><span class="va">string_db</span> <span class="op">&lt;-</span> <span class="va">STRINGdb</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>version <span class="op">=</span> <span class="st">"11.5"</span>, species <span class="op">=</span> <span class="fl">9606</span>, score_threshold <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Map STRING identifiers to gene identifiers in our DEA results dataframe</span></span>
<span><span class="va">mapped</span> <span class="op">&lt;-</span> <span class="va">string_db</span><span class="op">$</span><span class="fu">map</span><span class="op">(</span><span class="va">results</span>, <span class="st">"Protein.Names"</span>, removeUnmappedRows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Only map the proteins of significance</span></span>
<span><span class="va">de_proteins_mapped</span> <span class="op">&lt;-</span> <span class="va">mapped</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot network of significant proteins</span></span>
<span><span class="va">string_db</span><span class="op">$</span><span class="fu">plot_network</span><span class="op">(</span><span class="va">de_proteins_mapped</span><span class="op">$</span><span class="va">STRING_id</span><span class="op">)</span></span></code></pre>
</div>
<p><em>Note: if the above code takes too long to load, you can download
an RData that contains the <code>string_db</code> and
<code>mapped</code> functions <a href="https://github.com/egmg726/intro-to-proteomics/blob/main/episodes/data/string_db.RData" class="external-link">here</a>.</em></p>
<figure><img src="fig/05EnrichmentAnalysis-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="discussion1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div class="callout-inner">
<div class="callout-content">
<p>How many clusters are there?</p>
<p>How many of these interactions are predicted versus experimentally
validated? (See image below for legend)</p>
</div>
</div>
</div>
<p>You can explore how these proteins interact in an interactive setting
by going to <a href="https://string-db.org/" class="external-link">https://string-db.org/</a>
and further exploring your proteins of interest.</p>
<figure><img src="fig/05enrichmentanalysis_stringdb.png" alt="Network plot for PRTN3 from string-db.org" class="figure mx-auto d-block"><div class="figcaption">Network plot for PRTN3 from string-db.org</div>
</figure><div id="more-data-exploration-using-stringdb" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="more-data-exploration-using-stringdb" class="callout-inner">
<h3 class="callout-title">More data exploration using STRINGdb</h3>
<div class="callout-content">
<p>There are many ways of exploring your data using the STRING database
- too many to cover in one tutorial!</p>
<p>You can learn more by reading the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/STRINGdb/inst/doc/STRINGdb.pdf" class="external-link">vignette</a>
or inspecting available functions within the <code>STRINGdb</code>
package by running:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">STRINGdb</span><span class="op">$</span><span class="fu">methods</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] ".objectPackage"                      ".objectParent"
 [3] "add_diff_exp_color"                  "add_proteins_description"
 [5] "benchmark_ppi"                       "benchmark_ppi_pathway_view"
 [7] "callSuper"                           "copy"
 [9] "enrichment_heatmap"                  "export"
[11] "field"                               "get_aliases"
[13] "get_annotations"                     "get_bioc_graph"
[15] "get_clusters"                        "get_enrichment"
[17] "get_graph"                           "get_homologs"
[19] "get_homologs_besthits"               "get_homology_graph"
[21] "get_interactions"                    "get_link"
[23] "get_neighbors"                       "get_paralogs"
[25] "get_pathways_benchmarking_blackList" "get_png"
[27] "get_ppi_enrichment"                  "get_ppi_enrichment_full"
[29] "get_proteins"                        "get_pubmed"
[31] "get_pubmed_interaction"              "get_subnetwork"
[33] "get_summary"                         "get_term_proteins"
[35] "getClass"                            "getRefClass"
[37] "import"                              "initFields"
[39] "initialize"                          "load"
[41] "load_all"                            "map"
[43] "mp"                                  "plot_network"
[45] "plot_ppi_enrichment"                 "post_payload"
[47] "ppi_enrichment"                      "remove_homologous_interactions"
[49] "set_background"                      "show"
[51] "show#envRefClass"                    "trace"
[53] "untrace"                             "usingMethods"                       </code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Read more about STRING </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><em>Szklarczyk, D., Nastou, K., Koutrouli, M., Kirsch, R., Mehryary,
F., Hachilif, R., Hu, D., Peluso, M. E., Huang, Q., Fang, T., Doncheva,
N. T., Pyysalo, S., Bork, P., Jensen, L. J., &amp; von Mering, C.
(2025). The STRING database in 2025: protein networks with
directionality of regulation. Nucleic acids research, 53(D1), D730–D737.
<a href="https://doi.org/10.1093/nar/gkae1113" class="external-link uri">https://doi.org/10.1093/nar/gkae1113</a></em></p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="functional-enrichment-analysis-go-and-kegg">Functional enrichment analysis (GO and KEGG)<a class="anchor" aria-label="anchor" href="#functional-enrichment-analysis-go-and-kegg"></a>
</h2>
<hr class="half-width">
<p>Using the significant DE proteins, we can perform <strong>functional
enrichment analysis</strong> to determine whether certain biological
processes (BP), cellular components (CC), or molecular functions (MF)
are over-represented.</p>
<p>Two common types of enrichment exploration are:</p>
<ul>
<li>
<strong>Gene Ontology (GO)</strong>: describes BPs, CCs, amd
MFs</li>
<li>
<strong>KEGG pathways</strong>: curated molecular interactions</li>
</ul>
<div class="section level3">
<h3 id="go-enrichment-analysis">GO enrichment analysis<a class="anchor" aria-label="anchor" href="#go-enrichment-analysis"></a>
</h3>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Subset genes from results table</span></span>
<span><span class="va">de_proteins</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert UniProt IDs to Entrez IDs for enrichment analysis</span></span>
<span><span class="va">converted</span> <span class="op">&lt;-</span> <span class="fu">bitr</span><span class="op">(</span><span class="va">de_proteins</span><span class="op">$</span><span class="va">Protein.Group</span>,</span>
<span>                  fromType <span class="op">=</span> <span class="st">"UNIPROT"</span>,</span>
<span>                  toType <span class="op">=</span> <span class="st">"ENTREZID"</span>,</span>
<span>                  OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in bitr(de_proteins$Protein.Group, fromType = "UNIPROT", toType =
"ENTREZID", : 7.69% of input gene IDs are fail to map...</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GO Biological Process enrichment</span></span>
<span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">converted</span><span class="op">$</span><span class="va">ENTREZID</span>,</span>
<span>                OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,</span>
<span>                ont <span class="op">=</span> <span class="st">"ALL"</span>, <span class="co"># looking at all 3 subontologies</span></span>
<span>                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                readable <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show plot</span></span>
<span><span class="fu">dotplot</span><span class="op">(</span><span class="va">ego</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/05EnrichmentAnalysis-rendered-unnamed-chunk-5-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
<div class="section level3">
<h3 id="kegg-pathway-enrichment-analysis">KEGG pathway enrichment analysis<a class="anchor" aria-label="anchor" href="#kegg-pathway-enrichment-analysis"></a>
</h3>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># KEGG pathway enrichment</span></span>
<span><span class="va">ekegg</span> <span class="op">&lt;-</span> <span class="fu">enrichKEGG</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">converted</span><span class="op">$</span><span class="va">ENTREZID</span>,</span>
<span>                    organism <span class="op">=</span> <span class="st">'hsa'</span>,</span>
<span>                    pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Reading KEGG annotation online: "https://rest.kegg.jp/link/hsa/pathway"...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/hsa"...</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dotplot</span><span class="op">(</span><span class="va">ekegg</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/05EnrichmentAnalysis-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>How do these figures compare to <a href="https://mdpi-res.com/biomedicines/biomedicines-12-00333/article_deploy/html/images/biomedicines-12-00333-g004.png" class="external-link">Figure
4</a> in the original paper?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>While there are many similar pathways, the exact pathways are not
identical. This is likely due to the fact we are using a subset of the
larger dataset in this tutorial.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>Differential expression analysis identifies statistically
significant changes in protein abundance between conditions, but does
not necessarily tell us the biological significance of such identified
proteins.</li>
<li>Network analysis (STRING) can reveal physical or functional
interactions among DE proteins.</li>
<li>Enrichment analysis (GO/KEGG) can link those proteins to known
biological processes or pathways.</li>
<li>Using multiple approaches can help you interpret proteomics data to
measure reliability and consistency of results.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://melbournebioinformatics.github.io/intro-to-proteomics/" class="external-link">Source</a></p>
        <p>
        <a href="citation.html">Cite</a>
        | <a href="mailto:bioinformatics-training@unimelb.edu.au">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors.</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7)</a></p>
		</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "proteomics, mass spectrometry, DIA, DIA-NN, limpa, differential expression analysis, enrichment analysis, network analysis",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/aio.html",
  "dateCreated": "2025-10-30",
  "dateModified": "2025-11-14",
  "datePublished": "2025-11-14"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

