<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Introduction to Proteomics: Processing data with DIA-NN</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/uom/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/uom/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/uom/favicon-16x16.png"><link rel="manifest" href="../favicons/uom/site.webmanifest"><link rel="mask-icon" href="../favicons/uom/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="../assets/images/uom-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../02diann.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="../assets/images/uom-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Introduction to Proteomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Introduction to Proteomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="contact.html">Contact</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Introduction to Proteomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 23%" class="percentage">
    23%
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: 23%" aria-valuenow="23" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../02diann.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01dataset.html">1. Exploring the dataset</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        2. Processing data with DIA-NN
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#processing-proteomics-data">Processing proteomics data</a></li>
<li><a href="#dia-nn-workflow">DIA-NN workflow</a></li>
<li><a href="#dia-nn-output">DIA-NN output</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03datacleaning.html">3. Data cleaning with R</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04DEA.html">4. Differential expression analysis (DEA)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05EnrichmentAnalysis.html">5. Network and Enrichment Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="contact.html">Contact</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/01dataset.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/03datacleaning.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/01dataset.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Exploring the
        </a>
        <a class="chapter-link float-end" href="../instructor/03datacleaning.html" rel="next">
          Next: Data cleaning with R...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Processing data with DIA-NN</h1>
        <p>Last updated on 2025-11-12 |

        <a href="https://github.com/julia-broadbent/intro-to-proteomics/edit/main/episodes/02diann.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 15 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can I process my output files from a mass spectrometry-based
proteomics experiment?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Learn how to process proteomics data using DIA-NN.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="processing-proteomics-data">Processing proteomics data<a class="anchor" aria-label="anchor" href="#processing-proteomics-data"></a></h2>
<hr class="half-width"><p>There are many software options for processing mass
spectrometry-based proteomics output. These software match mass spectra
against an empirical or predicted spectral library to <strong>identify
peptides</strong> and <strong>infer the protein</strong> to which they
belong.</p>
<p>Some software you may have heard of include Spectronaut, Fragpipe,
MaxQuant and DIA-NN. Each software have slightly different functionality
and attributes, and are frequently updated. In this tutorial, we are
demonstrating the use of <strong>DIA-NN</strong>, a software designed to
process DIA proteomics data.</p>
<figure><img src="../fig/02diann_peptideMatching.png" alt="Proteomics software like DIA-NN match observed spectra against a spectral library to identify peptides and assign them to a protein or protein group. Source: Galaxy Training Network" class="figure mx-auto d-block"><div class="figcaption">Proteomics software like DIA-NN match observed
spectra against a spectral library to identify peptides and assign them
to a protein or protein group. Source: <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/introduction/slides.html#15" class="external-link">Galaxy
Training Network</a>
</div>
</figure><p><br></p>
<div class="section level3">
<h3 id="why-dia-nn">Why DIA-NN?<a class="anchor" aria-label="anchor" href="#why-dia-nn"></a></h3>
<p>Key features of DIA-NN software include:</p>
<ul><li>
<strong>Neural network–based scoring:</strong> Improves
identification confidence.</li>
<li>
<strong>Interference correction:</strong> Enhances quantification
accuracy.</li>
<li>
<strong>Library-free analysis:</strong> Can work without a spectral
library (generating one directly from DIA data).</li>
<li>
<strong>High speed and efficiency:</strong> Suitable for large-scale
datasets.</li>
<li>
<strong>Cross-run normalization:</strong> Ensures consistent
quantification across experiments.</li>
<li>
<strong>Command-line executable:</strong> DIA-NN can be run from a
graphical user interface (GUI) or directly from the command line, making
it easy to integrate into an automated pipeline.</li>
</ul><p>DIA-NN is free to download. Detailed operating instructions and a
link to install the latest version of DIA-NN can be found on the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">GitHub
page</a>.</p>
</div>
</section><section><h2 class="section-heading" id="dia-nn-workflow">DIA-NN workflow<a class="anchor" aria-label="anchor" href="#dia-nn-workflow"></a></h2>
<hr class="half-width"><div id="important-note" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="important-note" class="callout-inner">
<h3 class="callout-title">Important Note</h3>
<div class="callout-content">
<p>Due to the amount of time it will take to process the data, you do
not need to install or operate DIA-NN for the purposes of this tutorial.
Pre-processed data is provided, which has been prepared following the
procedures described below.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="part-1-creating-a-spectral-library">Part 1: Creating a spectral library<a class="anchor" aria-label="anchor" href="#part-1-creating-a-spectral-library"></a></h3>
<p>DIA-NN analysis consists of two steps. The first step is to
<strong>generate a predicted spectral library</strong> for your
experimental organism.</p>
<ol style="list-style-type: decimal"><li>
<p><strong>Download a sequence database</strong> from <a href="https://www.uniprot.org/uniprotkb?query=Human&amp;facets=model_organism%3A9606%2Creviewed%3Atrue" class="external-link">UniProt</a>.</p>
<ol style="list-style-type: lower-alpha"><li><p>Go to <a href="https://www.uniprot.org/uniprotkb?query=" class="external-link uri">https://www.uniprot.org/uniprotkb?query=</a>*</p></li>
<li><p>Select the <code>relevant organism</code> for your sample and
<code>Reviewed (Swiss-Prot)</code> status to filter results.</p></li>
<li><p>Click <code>Download</code></p></li>
<li><p>Change the format to
<code>FASTA (canonical and isoform)</code></p></li>
<li><p>Select <code>No</code> under Compressed and download the FASTA
file</p></li>
</ol></li>
</ol><div id="for-specific-searches" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="for-specific-searches" class="callout-inner">
<h3 class="callout-title">FOR SPECIFIC SEARCHES</h3>
<div class="callout-content">
<p>If you need to search for a variant protein or peptide that differs
from the standard organism proteome, you can manually edit the protein
sequences in the UniProt FASTA in a text editor.</p>
</div>
</div>
</div>
<ol start="2" style="list-style-type: decimal"><li><p><strong>Download a contaminant FASTA file</strong> from <a href="https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics/tree/main" class="external-link">here</a>.
For most studies, the Universal Contaminants.fasta will be appropriate,
but you may prefer to use one of the sample-type specific contaminant
FASTAs depending on your experiment. <strong><em>More information about
this resource and contaminants will be discussed in the next
lesson.</em></strong></p></li>
<li><p><strong>Open the DIA-NN</strong> GUI or a command line interface.
If there are multiple versions of DIA-NN installed on your device,
ensure to update <code>DIA-NN exe</code> with the file path to the
version you wish to use.</p></li>
<li><p><strong>Add the UniProt and Contaminant FASTA files</strong> you
just downloaded.</p></li>
</ol><div id="caution1" class="callout caution">
<div class="callout-square">
<i class="callout-icon" data-feather="alert-triangle"></i>
</div>
<span class="callout-header">Caution</span>
<div class="callout-inner">
<div class="callout-content">
<p>At the time of writing, DIA-NN 2.2.0 cannot read or write files saved
in an external drive. Please ensure all input and output files are saved
on a local drive to avoid error.</p>
</div>
</div>
</div>
<ol start="5" style="list-style-type: decimal"><li><p>Examples are included below for the additional <strong>settings
we recommend changing from the default</strong>. Depending on your
experiment and research goals, you may wish to change additional
settings from the default - read more in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings" class="external-link">DIA-NN
documentation</a>.</p></li>
<li><p><strong>Run</strong>. DIA-NN will output a predicted spectral
library in the <code>.speclib</code> file type.</p></li>
</ol><figure><img src="../fig/02diann_DIANNgui_speclib.png" alt="Example of the DIA-NN GUI used to generate a predicted spectral library. Settings which have been altered from the default are highlighted in red." class="figure mx-auto d-block"><div class="figcaption">Example of the DIA-NN GUI used to generate a
predicted spectral library. Settings which have been altered from the
default are highlighted in red.</div>
</figure><div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">C:\DIA-NN\2.2.0\diann.exe</span> <span class="kw">`</span> <span class="co"># Replace with file path to DIA-NN on your device</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">--lib</span> <span class="st">""</span> <span class="kw">`</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">--out</span> <span class="st">"C:\path\to\output\report.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with file path to output report.parquet</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">--out-lib</span> <span class="st">"C:\path\to\output\report-lib.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with file path to output report-lib.parquet</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\UniprotFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path to Uniprot FASTA</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">--fasta</span> <span class="st">"C:\path\to\ContaminantsFasta\.fasta"</span> <span class="kw">`</span> <span class="co"># Replace with file path to contaminants FASTA</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">--threads</span> 30 <span class="kw">`</span> <span class="co"># Replace with the number of threads to use for this analysis</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">--verbose</span> 1 <span class="kw">`</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">--qvalue</span> 0.01 <span class="kw">`</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="ex">--matrices</span>  <span class="kw">`</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="ex">--gen-spec-lib</span> <span class="kw">`</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="ex">--predictor</span> <span class="kw">`</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="ex">--reannotate</span> <span class="kw">`</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="ex">--fasta-search</span> <span class="kw">`</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="ex">--min-fr-mz</span> 200 <span class="kw">`</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="ex">--max-fr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="ex">--met-excision</span> <span class="kw">`</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="ex">--min-pep-len</span> 7 <span class="kw">`</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="ex">--max-pep-len</span> 30 <span class="kw">`</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="ex">--min-pr-mz</span> 300 <span class="kw">`</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="ex">--max-pr-mz</span> 1800 <span class="kw">`</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="ex">--min-pr-charge</span> 1 <span class="kw">`</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="ex">--max-pr-charge</span> 4 <span class="kw">`</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="ex">--cut</span> K<span class="pp">*</span>,R<span class="pp">*</span> <span class="kw">`</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="ex">--missed-cleavages</span> 1 <span class="kw">`</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="ex">--unimod4</span> <span class="kw">`</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="ex">--rt-profiling</span> </span></code></pre>
</div>
<p>This spectral library can be re-used for all analyses of samples from
the same organism. It is recommended to re-download the UniProt FASTA
and re-run this step approximately once a month to ensure your library
is up-to-date.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="part-2-analysis">Part 2: Analysis<a class="anchor" aria-label="anchor" href="#part-2-analysis"></a></h3>
<p>The second step is to <strong>search your raw files (mass
spectrometry output files)</strong> against the spectral library.</p>
<ol style="list-style-type: decimal"><li><p><strong>Open the DIA-NN</strong> GUI or a command line interface.
If there are multiple versions of DIA-NN installed on your device,
ensure to update <code>DIA-NN exe</code> with the file path to the
version you wish to use.</p></li>
<li>
<p><strong>Upload your raw files</strong>.</p>
<ul><li>If using the GUI, under <code>Input</code> select <code>Raw</code>
then select all of the files you wish to search.</li>
<li>If using the command line, you can use <code>--f</code> multiple
times to list each file individually, or (recommended) use
<code>--dir [folder]</code> to instruct DIA-NN to process all raw files
saved in a particular folder.</li>
</ul></li>
<li><p><strong>Upload your spectral library</strong>.</p></li>
</ol><div id="empirical-vs-predicted-spectral-library" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="empirical-vs-predicted-spectral-library" class="callout-inner">
<h3 class="callout-title">Empirical vs predicted spectral library</h3>
<div class="callout-content">
<p>You may wish to generate an empirical spectral library from your
data, and use this for future analyses. An empirical library will be
smaller than a predicted library and therefore increase the speed of
analysis; however, may result in fewer identifications. We only
recommend using an empirical library if generated from high quality
data.</p>
<p>You can generate an empirical library from your samples by selecting
<strong>Generate spectral library</strong> and specifying the location
of its output.</p>
</div>
</div>
</div>
<ol start="4" style="list-style-type: decimal"><li>
<p><strong>Adjust settings</strong> according to your experiment and
research goals. Again, examples are included below for the additional
settings we recommend changing from the default. You may wish to change
additional settings from the default - read more in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings" class="external-link">DIA-NN
documentation</a>.</p>
<ul><li>In this example, the <strong>protease</strong> Trypsin was used to
generate peptides during sample preparation.</li>
<li>Leaving <strong>Mass accuracy</strong> and <strong>MS1
accuracy</strong> on the default 0 means DIA-NN will optimise these
parameters automatically for the first run in the experiment and then
reuse the optimised settings for other runs. The DIA-NN documentation
recommends different settings depending on the type of mass spectrometer
used to generate your data. In this example, data were generated on a
<strong>TripleTOF 6600</strong>, so we set Mass accuracy to 20.0 and MS1
accuracy to 12.0</li>
<li>Set the number of <strong>Threads</strong> to be no more than the
number of logical cores on your computer.</li>
<li>In this experiment, we have set the <strong>maximum number of
variable modifications</strong> to 0; however, it is common for this
setting to be relaxed in the range of 1 - 3.</li>
<li>DIA-NN’s default setting is to activate
<strong>match-between-runs</strong>. From the documentation: <em>In MBR
mode, DIA-NN does two passes over the data. During the first pass, it
creates an empirical spectral library from the data. During the second
pass, it reanalyses the experiment with this empirical library, which
may result in much improved identification numbers, data completeness
and quantification accuracy.</em>
</li>
</ul></li>
</ol><figure><img src="../fig/02diann_DIANNgui_analysis.png" alt="Example of the DIA-NN GUI used to analyse the data described in this tutorial. Settings which have been altered from the default are highlighted in red." class="figure mx-auto d-block"><div class="figcaption">Example of the DIA-NN GUI used to analyse the
data described in this tutorial. Settings which have been altered from
the default are highlighted in red.</div>
</figure><div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">C:\DIA-NN\2.2.0\diann.exe</span> <span class="kw">`</span> <span class="co"># Replace with file path to DIA-NN on your device</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">--dir</span> <span class="st">"C:\path\to\folder\of\raw\files"</span> <span class="kw">`</span> <span class="co"># Replace with file path to folder containing all raw files to be analysed</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">--lib</span> <span class="st">"C:\path\to\SpectralLibrary\report-lib.predicted.speclib"</span> <span class="kw">`</span> <span class="co"># Replace with file path to spectral library</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="ex">--out</span> <span class="st">"C:\path\to\output\report.parquet"</span> <span class="kw">`</span> <span class="co"># Replace with file path to output report.parquet</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="ex">--threads</span> 30 <span class="kw">`</span> <span class="co"># Replace with the number of threads to use for this analysis</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="ex">--verbose</span> 1 <span class="kw">`</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="ex">--qvalue</span> 0.01 <span class="kw">`</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="ex">--matrices</span>  <span class="kw">`</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">--unimod4</span> <span class="kw">`</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="ex">--mass-acc</span> 20 <span class="kw">`</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="ex">--mass-acc-ms1</span> 12.0 <span class="kw">`</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="ex">--reanalyse</span> <span class="kw">`</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="ex">--rt-profiling</span> </span></code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>If you want to run DIA-NN from the command line but are unsure how to
write the code, you can update your settings in the GUI, click ‘Run’,
and copy the commands output in the log. That’s how we got the example
code above! <strong>Can you match each line of code to the corresponding
setting in the GUI?</strong></p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Check the description of available commands in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#command-line-reference" class="external-link">DIA-NN
documentation</a></p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="dia-nn-output">DIA-NN output<a class="anchor" aria-label="anchor" href="#dia-nn-output"></a></h2>
<hr class="half-width"><p>DIA-NN output includes several files, including:</p>
<table class="table"><colgroup><col width="21%"><col width="78%"></colgroup><thead><tr class="header"><th>Output file</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><strong>log.txt</strong></td>
<td>A log of what DIA-NN has run. You should always use
<code>ctrl + F</code> to check for any ‘ERROR’ or ‘WARNING’ messages
issued by DIA-NN during your run.</td>
</tr><tr class="even"><td><strong>.parquet</strong></td>
<td>This is what we will be using for subsequent analyses in this
tutorial.</td>
</tr><tr class="odd"><td><strong>pg_matrix.tsv</strong></td>
<td>Matrix of protein values generated by DIA-NN.</td>
</tr><tr class="even"><td><strong>pr_matrix.tsv</strong></td>
<td>Matrix of peptide values generated by DIA-NN.</td>
</tr><tr class="odd"><td><strong>trends.pdf</strong></td>
<td>A PDF report listing the number of peptides, proteins, and other
quality control measures detected for each sample.</td>
</tr></tbody></table><div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>If DIA-NN is run through the command line rather than the GUI, the
trends.pdf and runs.pdf reports are not automatically generated. Using
the documentation provided in the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings" class="external-link">DIA-NN
GitHub page</a>, can you work out how to generate this report via the
command line?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Run the <code>diann-stats.py</code> Python script provided when you
download DIA-NN.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">python</span> C:<span class="dt">\D</span>IA-NN<span class="dt">\2</span>.2.0<span class="dt">\d</span>iann-stats.py C:<span class="dt">\p</span>ath<span class="dt">\t</span>o<span class="dt">\r</span>eport.parquet</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>We have now processed our raw files, identifying the peptides present
in our samples and inferring the proteins to which they belong.</p>
<div id="learn-more-about-diann" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="learn-more-about-diann" class="callout-inner">
<h3 class="callout-title">Learn more about DIANN</h3>
<div class="callout-content">
<p>For more background information, see the original publication:<br><em>Demichev, V., et al. (2020). DIA-NN: neural networks and
interference correction enable deep proteome coverage in high
throughput. Nature Methods.</em> <a href="https://doi.org/10.1038/s41592-019-0638-x" class="external-link">https://doi.org/10.1038/s41592-019-0638-x</a>.</p>
<p>For further details about the different DIA-NN settings, see the <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">documentation</a>.</p>
<p>If you come across any issues, the creators are fairly active at
responding to <a href="https://github.com/vdemichev/DiaNN/issues" class="external-link">issues</a> and <a href="https://github.com/vdemichev/DiaNN/discussions" class="external-link">discussions</a> on
their GitHub.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>Raw files from a mass-spectrometry based DIA proteomics experiment
can be processed using DIA-NN (peptide identification and protein
inference).</li>
<li>The DIA-NN workflow has two key steps: generated a predicted
spectral library, and processing raw files.</li>
<li>The <a href="https://github.com/vdemichev/DiaNN?tab=readme-ov-file" class="external-link">DIA-NN
documentation</a> contains detailed information about changing default
settings as appropriate for your experiment.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/01dataset.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/03datacleaning.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/01dataset.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Exploring the
        </a>
        <a class="chapter-link float-end" href="../instructor/03datacleaning.html" rel="next">
          Next: Data cleaning with R...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/julia-broadbent/intro-to-proteomics/edit/main/episodes/02diann.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/julia-broadbent/intro-to-proteomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/julia-broadbent/intro-to-proteomics/" class="external-link">Source</a></p>
        <p>
        <a href="../citation.html">Cite</a>
        | <a href="mailto:bioinformatics-training@unimelb.edu.au">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors.</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/instructor/02diann.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "proteomics, mass spectrometry, DIA, DIA-NN, limpa, differential expression analysis, enrichment analysis, network analysis",
  "name": "Processing data with DIA-NN",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/instructor/02diann.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/instructor/02diann.html",
  "dateCreated": "2025-10-30",
  "dateModified": "2025-11-12",
  "datePublished": "2025-11-12"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

