<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Introduction to Proteomics: Data cleaning with R</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/uom/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/uom/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/uom/favicon-16x16.png"><link rel="manifest" href="../favicons/uom/site.webmanifest"><link rel="mask-icon" href="../favicons/uom/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="../assets/images/uom-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../03datacleaning.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="This workshop will introduce participants to proteomic data processing, cleaning, and analysis, with a focus on label-free, bottom-up, DIA, mass-spectrometry based quantitative proteomics." src="../assets/images/uom-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Introduction to Proteomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Introduction to Proteomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="contact.html">Contact</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Introduction to Proteomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 45%" class="percentage">
    45%
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: 45%" aria-valuenow="45" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../03datacleaning.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01dataset.html">1. Exploring the dataset</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02diann.html">2. Processing data with DIA-NN</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Data cleaning with R
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#load-processed-data-and-sample-annotation">Load processed data and sample annotation</a></li>
<li><a href="#quality-filtering">Quality filtering</a></li>
<li><a href="#removing-contaminants">Removing contaminants</a></li>
<li><a href="#protein-quantification">Protein quantification</a></li>
<li><a href="#dimensionality-reduction-and-qc">Dimensionality Reduction and QC</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04DEA.html">4. Differential expression analysis (DEA)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05EnrichmentAnalysis.html">5. Network and Enrichment Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="contact.html">Contact</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/02diann.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/04DEA.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/02diann.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Processing data with
        </a>
        <a class="chapter-link float-end" href="../instructor/04DEA.html" rel="next">
          Next: Differential...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Data cleaning with R</h1>
        <p>Last updated on 2025-11-12 |

        <a href="https://github.com/julia-broadbent/intro-to-proteomics/edit/main/episodes/03datacleaning.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do we convert our processed proteomics data (DIA-NN output) into
an analysis-ready format?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the steps required to clean proteomics data and generate
analysis-ready protein values using <code>limpa</code>.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="load-processed-data-and-sample-annotation">Load processed data and sample annotation<a class="anchor" aria-label="anchor" href="#load-processed-data-and-sample-annotation"></a></h2>
<hr class="half-width"><p>After processing raw proteomics data, further cleaning and analysis
can be conducted using <strong>R</strong>, <strong>Python</strong>, or
by importing data to other platforms like <strong>Perseus</strong> or
<strong>Skyline</strong>.</p>
<p>In this tutorial, we will be teaching you one method of cleaning and
analysing proteomics data using R. <strong>To follow along the remainder
of this tutorial, you should open a new R script in RStudio and run the
example code provided.</strong></p>
<div id="the-limpa-package" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="the-limpa-package" class="callout-inner">
<h3 class="callout-title">The limpa package</h3>
<div class="callout-content">
<p>In this tutorial, we are using the <code>limpa</code> package to load
and clean our data. This package was published in 2025 by the <a href="https://github.com/SmythLab/limpa" class="external-link">Smyth Lab</a>, based at the
Walter and Eliza Hall Institute of Medical Research (WEHI).</p>
<p>For background information, see the original publications:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">-</span> Li, M., Cobbold, S. A., <span class="kw">&amp;</span> <span class="ex">Smyth,</span> G. K. <span class="er">(</span><span class="ex">2025</span><span class="kw">)</span><span class="bu">.</span> Quantification and differential analysis of mass spectrometry proteomics data with probabilistic recovery of information from missing values. bioRxiv, 2025.2004.2028.651125. https://doi.org/10.1101/2025.04.28.651125 </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">-</span> Li, M., <span class="kw">&amp;</span> <span class="ex">Smyth,</span> G. K. <span class="er">(</span><span class="ex">2023</span><span class="kw">)</span><span class="bu">.</span> Neither random nor censored: estimating intensity-dependent probabilities for missing values in label-free proteomics. Bioinformatics, 39<span class="er">(</span><span class="ex">5</span><span class="kw">)</span><span class="bu">.</span> https://doi.org/10.1093/bioinformatics/btad200 </span></code></pre>
</div>
<p>You can also find further information in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">vignette</a>
or <a href="https://bioconductor.org/packages/release/bioc/manuals/limpa/man/limpa.pdf" class="external-link">reference
manual</a> accessed via the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">Bioconductor
page</a>.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a></h3>
<p>First, load the required R packages.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">limpa</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">EnhancedVolcano</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">STRINGdb</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">clusterProfiler</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">arrow</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">rpx</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">stringr</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-dia-nn-output">Load DIA-NN output<a class="anchor" aria-label="anchor" href="#load-dia-nn-output"></a></h3>
<p><strong>Load the peptide-level data</strong> processed by DIA-NN
(<code>.parquet</code> file), <strong>filtering observations</strong> by
quality metrics as recommended in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">limpa
vignette</a>. The peptide values will automatically be <strong>converted
to log2</strong> upon import.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">readDIANN</span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/MBIntroToProteomics.parquet'</span>,format<span class="op">=</span><span class="st">"parquet"</span>,</span>
<span>                       q.columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Q.Value"</span>,<span class="st">"Lib.Q.Value"</span>,<span class="st">"Lib.PG.Q.Value"</span><span class="op">)</span>,</span>
<span>                       q.cutoffs <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Length of q-value columns does not match with length of q-value cutoffs. Use q.cutoffs[1] for all columns.</code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>These quality filtering metrics are recommended in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limpa/inst/doc/limpa.html" class="external-link">limpa
vignette</a> for data searched in DIA-NN <strong>with
match-between-runs</strong>.</p>
<p>What <code>q.columns</code> are recommended for data searched
<strong>without MBR</strong>?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">readDIANN</span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/MBIntroToProteomics.parquet'</span>,format<span class="op">=</span><span class="st">"parquet"</span>,</span>
<span>                       q.columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Q.Value"</span>,<span class="st">"Global.Q.Value"</span>,<span class="st">"Global.PG.Q.Value"</span><span class="op">)</span>,</span>
<span>                       q.cutoffs <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Let’s look at the structure of the data.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "E"     "genes"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see, our data has been imported as a limma EList object,
with components <strong>‘E’ (log2 peptide matrix)</strong> and
<strong>‘genes’ (feature annotation)</strong> containing information
about each peptide.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="load-sample-annotation">Load sample annotation<a class="anchor" aria-label="anchor" href="#load-sample-annotation"></a></h3>
<p>We also need to <strong>import information about our samples</strong>
for later analyses.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples_stool</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">readxl</span><span class="fu">::</span><span class="fu">read_excel</span><span class="op">(</span><span class="st">'data/SampleAnnotation.xlsx'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  file_order Sample_id-new Batch Class
1         33       B1_29_c    B1   CDr
2         93       B1_87_e    B1   UCr
3         71       B1_66_e    B1   UCr
4          9       B1_08_b    B1   aCD
5         26       B1_22_c    B1   CDr
6         31       B1_27_c    B1   CDr
                                                                                                                                         file_name
1 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_067-29.mzML
2    D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_187.mzML
3 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_143-66.mzML
4  D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_019-8.mzML
5 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_053-22.mzML
6 D:\\Elmira\\IBD\\01-MSconverter\\00-DIA_NN input-3 batches\\mix.mzMLpeakpicking(batch 1,2)\\Mix\\20201016_UdSjfb_20190115_freshstool_063-27.mzML</code></pre>
</div>
<p>We need to clean our sample annotation data to add an identifying
column which <strong>matches the sample column names in our peptide
matrix.</strong></p>
<p>Given we are only using a subset of the main dataset in this
tutorial, we can also filter the sample annotation dataframe to
<strong>only include rows pertaining to the samples in our
subset.</strong></p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create sample_name column that matches peptide matrix column names</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">"\\\\"</span>, <span class="st">"/"</span>, <span class="va">samples_stool</span><span class="op">$</span><span class="va">file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span> <span class="op">&lt;-</span> <span class="fu">basename</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span>
<span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">".mzML"</span>, <span class="st">""</span>,<span class="va">samples_stool</span><span class="op">$</span><span class="va">fixed_file_name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to only samples present in our dataset</span></span>
<span><span class="va">samples_stool</span> <span class="op">&lt;-</span> <span class="va">samples_stool</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">%in%</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>We can attach our sample annotation data to the EList object for
later analyses.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure sample row order matches column order of the peptide matrix</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">samples_stool</span><span class="op">[</span><span class="va">samples_stool</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">%in%</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">sample_name</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span>,<span class="fu">rownames</span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Add experimental metadata to EList</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'Class'</span>, <span class="st">'Batch'</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="quality-filtering">Quality filtering<a class="anchor" aria-label="anchor" href="#quality-filtering"></a></h2>
<hr class="half-width"><p>After reading in the data, it is common to conduct additional
filtering to improve the quality of your data and assist with downstream
interpretation of results.</p>
<ul><li>
<strong>Filtering out non-proteotypic peptides:</strong> Removes
peptides that belong to more than one protein.</li>
<li>
<strong>Filtering out compound proteins:</strong> Removes peptides
belonging to compound protein groups consisting of multiple proteins
separated by “;” delimiters.</li>
<li>
<strong>Filtering out singleton peptides:</strong> Removes peptides
that are the only peptide belonging to that protein (i.e. keeps only
proteins with multiple peptides).</li>
</ul><div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterNonProteotypicPeptides</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterCompoundProteins</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">)</span></span>
<span><span class="va">y.peptide</span> <span class="op">&lt;-</span> <span class="fu">filterSingletonPeptides</span><span class="op">(</span><span class="va">y.peptide</span>, min.n.peptides <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<p>These filtering steps are optional; if they are not run,
<code>limpa</code> will still be able to conduct protein quantification.
For more information about these peptide-level filters and whether they
are appropriate for your experiment, please consult the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">limpa
documentation</a>.</p>
</section><section><h2 class="section-heading" id="removing-contaminants">Removing contaminants<a class="anchor" aria-label="anchor" href="#removing-contaminants"></a></h2>
<hr class="half-width"><p>Contaminant background signals can influence mass spectrometry-based
proteomics data. It is essential to <strong>remove contaminant
proteins</strong> from the sample preparation process to avoid
confounding results.</p>
<p><a href="https://doi.org/10.1021/acs.jproteome.2c00145" class="external-link">Frankenfield
et al. (2022)</a> created a new set of sample-type specific and
universal contaminant FASTA files which they showed reduced false
identifications, increased protein identifications and did not influence
protein quantification for DIA workflows. These FASTA files updated upon
widely used contaminant protein lists from MaxQuant and the common
Repository of Adventitious Proteins (cRAP).</p>
<p>In the previous lesson, we incorporated the universal contaminants
FASTA into our spectral library. This FASTA is structured such that all
contaminant proteins are named beginning with <strong>‘Cont_’</strong>,
which means we can simply remove any proteins from our analysis
containing this string text in their name.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove contaminant proteins from the genes dataframe</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">Protein.Group</span>, <span class="st">"Cont_"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Match peptides in the matrix to only those remaining in the dataframe</span></span>
<span><span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span> <span class="op">&lt;-</span> <span class="va">y.peptide</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span>, <span class="op">]</span></span></code></pre>
</div>
<div id="more-about-contaminants" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="more-about-contaminants" class="callout-inner">
<h3 class="callout-title">More about contaminants</h3>
<div class="callout-content">
<p>You can <strong>download the contaminant FASTA files from <a href="https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics" class="external-link">GitHub</a>.</strong></p>
<p>If you are analysing data that has been processed without a
contaminant FASTA, you can also download an <strong>Excel spreadsheet
contaminant list</strong> from the same link and use this to identify
and remove contaminant proteins from your dataset.</p>
<p>For more information, see the original publication:</p>
<p>Frankenfield, A. M., Ni, J., Ahmed, M., &amp; Hao, L. (2022). Protein
Contaminants Matter: Building Universal Protein Contaminant Libraries
for DDA and DIA Proteomics. Journal of proteome research, 21(9),
2104–2113. <a href="https://doi.org/10.1021/acs.jproteome.2c00145" class="external-link uri">https://doi.org/10.1021/acs.jproteome.2c00145</a></p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="protein-quantification">Protein quantification<a class="anchor" aria-label="anchor" href="#protein-quantification"></a></h2>
<hr class="half-width"><p>Missing data has been a longstanding issue impacting the accuracy and
reproducibility of mass spectrometry-based proteomic analyses,
complicating downstream analyses. Most established imputation methods
are not designed to account for the specific attributes of proteomics
data, and there is no consensus in the literature for how missing values
should be managed in proteomics experiments. This presents an issue with
using the protein values directly generated by platforms like DIA-NN,
which contain many missing values.</p>
<p>In this tutorial, we are using the <code>dpcQuant()</code> method of
protein quantification from the <code>limpa</code> package, or
<em>Linear Models for Proteomics Data</em> <a href="https://doi.org/10.1101/2025.04.28.651125" class="external-link">(Li, Cobbold &amp;
Smyth, 2025)</a>. The <strong>detection probability curve (DPC)</strong>
models the relationship between peptide intensity and missingness,
showing that almost all missing values in label-free proteomics data are
not missing at random and should be accounted for <a href="https://doi.org/10.1093/bioinformatics/btad200" class="external-link">(Li &amp; Smyth,
2023)</a>. The <code>dpcQuant()</code> uses Bayesian statistical methods
based on peptide intensities and the DPC coefficients for a particular
dataset to generate a complete matrix of protein values without the need
for imputation. This method outputs a linear model object suitable for
downstream analysis with the <code>limma</code> package.</p>
<p>To learn more about the detection probability curve and the
<code>limpa</code> package, please read the original publications.</p>
<div class="section level3">
<h3 id="using-limpa-and-the-dpc-for-protein-quantification">Using limpa and the DPC for protein quantification<a class="anchor" aria-label="anchor" href="#using-limpa-and-the-dpc-for-protein-quantification"></a></h3>
<p>First, we will generate and visualise our <strong>detection
probability curve</strong>.</p>
<p>The DPC coefficients are related to the dataset and software used for
processing. A steeper slope means there is more statistical information
in the pattern of missing data. A slope closer to 0 means there is more
data missing at random.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate the DPC for our data</span></span>
<span><span class="va">dpcfit</span> <span class="op">&lt;-</span> <span class="fu">dpcON</span><span class="op">(</span><span class="va">y.peptide</span>, robust<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the intercept and slope of our DPC</span></span>
<span><span class="va">dpcfit</span><span class="op">$</span><span class="va">dpc</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>     beta0      beta1
-7.2678650  0.6033267 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the DPC</span></span>
<span><span class="fu">plotDPC</span><span class="op">(</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/03datacleaning-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Next, we can use the DPC to generate our log2 protein values. This
function requires the following input:</p>
<ul><li>a limma EList object, or a numeric matrix, of peptide-level log2
expression values, with samples as columns and peptides as rows
(<code>y.peptide</code>)</li>
<li>vector of protein IDs (<code>"Protein.Names"</code>)</li>
<li>numeric vector providing the slope and intercept of the DPC
(<code>dpcfit</code>)</li>
</ul><div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y.protein</span> <span class="op">&lt;-</span> <span class="fu">dpcQuant</span><span class="op">(</span><span class="va">y.peptide</span>, <span class="st">"Protein.Names"</span>, dpc<span class="op">=</span><span class="va">dpcfit</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Estimating hyperparameters ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Quantifying proteins ...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Proteins: 280 Peptides: 2672</code></pre>
</div>
<p><em>Note: our dataset is quite small, so this process should only
take a few seconds. For larger datasets, it may take minutes or hours
for this command to run.</em></p>
<div id="alternate-versions-of-the-dpc" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="alternate-versions-of-the-dpc" class="callout-inner">
<h3 class="callout-title">Alternate versions of the DPC</h3>
<div class="callout-content">
<p>The base <code>dpc()</code> assumes observed log-intensities are
normally distributed.</p>
<p>A modified version, used in this tutorial,
<code>dpcON(robust=TRUE)</code> downweighs outliers.</p>
<p>Alternatively, <code>dpcCN()</code> assumes the complete set of
log-intensities are normally distributed. Using this method may
underestimate the DPC slope if there is a non-small proportion of
peptides missing by chance.</p>
<p>You can also choose your own slope by passing the argument
<code>dpc.slope=</code> to <code>dpcQuant()</code> instead of
<code>dpc=dpcfit</code>. Using this method, an intercept value will
automatically be generated. You can check this value by using
<code>estimateDPCIntercept()</code>. The authors recommend a slope of
0.6 - 0.8 is usually appropriate for data generated using DIA-NN.</p>
<p>You may wish to conduct further statistical analyses to determine
which method is most appropriate for your experiment; however, the
authors suggest that downstream analysis should not be significantly
affected by differing DPC coefficients.</p>
<p>You can read more about the differences between each version of the
DPC in the <a href="https://www.biorxiv.org/content/10.1101/2025.04.28.651125v1" class="external-link">paper</a>
or the <a href="https://bioconductor.org/packages/release/bioc/manuals/limpa/man/limpa.pdf" class="external-link">limpa
manual</a>.</p>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>The limpa package also provides a function to impute missing peptide
values and generate a complete peptide matrix using the DPC. Can you
search the <a href="https://bioconductor.org/packages/release/bioc/html/limpa.html" class="external-link">limpa
documentation</a> to identify the command used to impute and quantify
missing values in a rowwise manner?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p><code>dpcImpute()</code></p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="dpcquant-output">dpcQuant() output<a class="anchor" aria-label="anchor" href="#dpcquant-output"></a></h3>
<p>We have now generated an EList object for our protein values. Let’s
investigate the other components of this object:</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "E"       "genes"   "other"   "targets" "dpc"    </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">$</span><span class="va">other</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "n.observations" "standard.error"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">y.protein</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Protein.Group" "Protein.Names" "Genes"         "NPeptides"
[5] "PropObs"      </code></pre>
</div>
<table class="table"><colgroup><col width="28%"><col width="72%"></colgroup><thead><tr class="header"><th align="center">EList component</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td align="center"><strong>E</strong></td>
<td>A complete matrix of log2 protein values with no missing data, with
samples as columns and proteins as rows</td>
</tr><tr class="even"><td align="center"><strong>genes</strong></td>
<td>Feature annotation dataframe, with columns for: UniProt IDs
(Protein.Group), proteins (Protein.Names), gene names (Genes), number of
peptides used to quantify each protein (NPeptides), and the proportion
of possible observations used to generate each protein value
(PropObs)</td>
</tr><tr class="odd"><td align="center"><strong>other</strong></td>
<td>List of two matrices with columns and rows corresponding to protein
matrix E: number of peptides observed for each protein in each sample
(n.observations), and the standard error associated with each protein
value for each sample (standard.error)</td>
</tr><tr class="even"><td align="center"><strong>targets</strong></td>
<td>Sample annotation dataframe</td>
</tr><tr class="odd"><td align="center"><strong>dpc</strong></td>
<td>DPC intercept and slope coefficient used for protein
quantification</td>
</tr></tbody></table><div id="challenge3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Compare the first few rows of the protein (<code>E</code>),
<code>n.observations</code> and <code>standard.error</code> matrices.
Take a look at the <code>NPeptides</code> and <code>PropObs</code>
values for those proteins also. What do you notice about the
relationship between these quality metrics?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Proteins with smaller <code>NPeptides</code> and <code>PropObs</code>
values, and smaller <code>n.observations</code> for individual samples,
have higher standard error.</p>
<p>The more peptide data available for a particular protein in a
particular sample, the greater the confidence in the protein value, and
therefore the lower the standard error.</p>
<p>If you are interested in a particular protein, it is important to
look at these quality metrics for each of your samples.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="dimensionality-reduction-and-qc">Dimensionality Reduction and QC<a class="anchor" aria-label="anchor" href="#dimensionality-reduction-and-qc"></a></h2>
<hr class="half-width"><p>Let’s take a look at our dataset. The <code>plotMDSUsingSEs()</code>
function, or <em>Multidimensional Scaling Plot of Gene Expression
Profiles, Using Standard Errors</em>, is similar to
<code>plotMDS()</code> in the <code>limma</code> package, but takes
account of the standard errors generated by <code>dpcQuant()</code>.
This is essentially a PCA plot for the newly generated protein values
that takes into consideration their standard error.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/03datacleaning-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>That plot is extremely messy and doesn’t tell us much! Let’s clean it
up.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Factorise and set colors for class and batch</span></span>
<span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Class</span><span class="op">)</span></span>
<span><span class="va">Class.color</span> <span class="op">&lt;-</span> <span class="va">Class</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">hcl.colors</span><span class="op">(</span><span class="fu">nlevels</span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>, palette <span class="op">=</span> <span class="st">"cividis"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">y.peptide</span><span class="op">$</span><span class="va">targets</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">Batch.color</span> <span class="op">&lt;-</span> <span class="va">Batch</span></span>
<span><span class="fu">levels</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">hcl.colors</span><span class="op">(</span><span class="fu">nlevels</span><span class="op">(</span><span class="va">Batch</span><span class="op">)</span>, palette <span class="op">=</span> <span class="st">"Set 2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Class visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Class.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/03datacleaning-rendered-unnamed-chunk-15-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Batch visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, pch<span class="op">=</span><span class="fl">16</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/03datacleaning-rendered-unnamed-chunk-15-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Class (labels) and batch (colors) visualisation</span></span>
<span><span class="fu">plotMDSUsingSEs</span><span class="op">(</span><span class="va">y.protein</span>, labels<span class="op">=</span><span class="va">Class</span>, col<span class="op">=</span><span class="fu">as.character</span><span class="op">(</span><span class="va">Batch.color</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/03datacleaning-rendered-unnamed-chunk-15-3.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can already see a distinction between the aCD and Ctrl groups,
which will be explored further in the next lesson.</p>
<p>It also looks like there are some differences between the batches.
<strong>We should add batch as a covariate to our analyses in the next
lesson.</strong></p>
<div id="managing-batch-effects" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="managing-batch-effects" class="callout-inner">
<h3 class="callout-title">Managing batch effects</h3>
<div class="callout-content">
<p>Explaining the different methods of identifying, accounting and
correcting for batch effects in proteomics data is an entire tutorial
itself! For further information, we recommend the <a href="https://github.com/melbintgen/Batch-effect-management?tab=readme-ov-file" class="external-link">Melbourne
Integrative Genomics Workshop: Managing batch effects in biological
studies</a>.</p>
<p>If you are concerned about batch effects in your data, University of
Melbourne staff and students or affiliates can also <a href="https://mdhs.unimelb.edu.au/melbournebioinformatics/training-and-support/expert-advice" class="external-link">contact
us</a> for a free consultation.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="normalisation">Normalisation<a class="anchor" aria-label="anchor" href="#normalisation"></a></h3>
<p>Normalisation is another key element of the proteomics data cleaning
workflow. Normalisation techniques aim to remove non-biological
variation in the data, for example, variation due to sample loading or
batch effects.</p>
<p>The default RT-dependent normalisation applied during DIA-NN
processing is quite effective and often sufficient for DIA data;
however, each experimental environment and techniques will have a
different optimal approach. You may wish to apply an additional method
of normalisation to your data.</p>
<p>To learn more about normalisation, keep an eye on the <a href="https://www.eventbrite.com.au/o/melbourne-bioinformatics-13058846490" class="external-link">Melbourne
Bioinformatics Eventbrite page</a> for a normalisation workshop coming
in 2026.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> More about normalisation </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>To learn more about different normalisation techniques for proteomics
data, we recommend the following papers:</p>
<p>Peng, H., Wang, H., Kong, W., Li, J., &amp; Goh, W. W. B. (2024).
Optimizing differential expression analysis for proteomics data via
high-performing rules and ensemble inference. Nature communications,
15(1), 3922. <a href="https://doi.org/10.1038/s41467-024-47899-w" class="external-link uri">https://doi.org/10.1038/s41467-024-47899-w</a></p>
<p>Tseng, C. Y., Salguero, J. A., Breidenbach, J. D., Solomon, E.,
Sanders, C. K., Harvey, T., Thornhill, M. G., Palmisano, S. J., Sasiene,
Z. J., Blackwell, B. R., McBride, E. M., Luchini, K. A., LeBrun, E. S.,
Alvarez, M., Mach, P. M., Rivera, E. S., &amp; Glaros, T. G. (2025).
Evaluation of normalization strategies for mass spectrometry-based
multi-omics datasets. Metabolomics : Official journal of the Metabolomic
Society, 21(4), 98. <a href="https://doi.org/10.1007/s11306-025-02297-1" class="external-link uri">https://doi.org/10.1007/s11306-025-02297-1</a></p>
<p>Välikangas, T., Suomi, T., &amp; Elo, L. L. (2018). A systematic
evaluation of normalization methods in quantitative label-free
proteomics. Briefings in bioinformatics, 19(1), 1–11. <a href="https://doi.org/10.1093/bib/bbw095" class="external-link uri">https://doi.org/10.1093/bib/bbw095</a></p>
</div>
</div>
</div>
</div>
<p>Good work! We now have a clean dataset ready for analysis.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>Key elements of the proteomics data cleaning workflow include
quality filtering, removing contaminants, protein quantification,
managing batch effects, and normalisation.</li>
<li>We can use the <em>limpa</em> package in R to clean our peptide data
and generate a complete, analysis-ready protein matrix with
corresponding standard error values.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/02diann.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/04DEA.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/02diann.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Processing data with
        </a>
        <a class="chapter-link float-end" href="../instructor/04DEA.html" rel="next">
          Next: Differential...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/julia-broadbent/intro-to-proteomics/edit/main/episodes/03datacleaning.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/julia-broadbent/intro-to-proteomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/julia-broadbent/intro-to-proteomics/" class="external-link">Source</a></p>
        <p>
        <a href="../citation.html">Cite</a>
        | <a href="mailto:bioinformatics-training@unimelb.edu.au">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors.</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/instructor/03datacleaning.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "proteomics, mass spectrometry, DIA, DIA-NN, limpa, differential expression analysis, enrichment analysis, network analysis",
  "name": "Data cleaning with R",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/instructor/03datacleaning.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/instructor/03datacleaning.html",
  "dateCreated": "2025-10-30",
  "dateModified": "2025-11-12",
  "datePublished": "2025-11-12"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

