---
title: 'Processing data with DIA-NN'
teaching: 10
exercises: 5
---


:::::::::::::::::::::::::::::::::::::: questions 

- How can I process my output files from a mass spectrometry-based proteomics experiment? 

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Learn how to process proteomics data using DIA-NN

::::::::::::::::::::::::::::::::::::::::::::::::

## Processing proteomics data

There are many software options for processing mass spectrometry-based proteomics output. These software *match mass spectra* against an empirical or predicted spectral library to *identify peptides* and infer the protein to which they belong.

Some software you may have heard of include Spectronaut, Fragpipe, MaxQuant and DIA-NN. Each software have slightly different functionality and attributes, and are frequently updated. In this tutorial, we are demonstrating the use of **DIA-NN**, a software designed to process **DIA** proteomics data.

![Proteomics software like DIA-NN match observed spectra against a spectral library to identify peptides and assign them to a protein or protein group. Source: [Galaxy Training Network](https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/introduction/slides.html#15)](episodes/fig/02diann_peptideMatching.png)

### Why DIA-NN?

Key features of DIA-NN software include:

- **Neural networkâ€“based scoring:** Improves identification confidence.
- **Interference correction:** Enhances quantification accuracy.
- **Library-free analysis:** Can work without a spectral library (generating one directly from DIA data).
- **High speed and efficiency:** Suitable for large-scale datasets.
- **Cross-run normalization:** Ensures consistent quantification across experiments.
- **Command-line executable:** DIA-NN can be run from a graphical user interface (GUI) or directly from the command line, making it easy to integrate into an automated pipeline.

DIA-NN is free to download. Detailed operating instructions and a link to install the latest version of DIA-NN can be found on the [GitHub page](https://github.com/vdemichev/DiaNN?tab=readme-ov-file).

:::callout
# Important Note

Due to the amount of time it will take to process the data, you do not need to install or operate DIA-NN for the purposes of this tutorial. Pre-processed data is provided, which has been prepared following the procedures described below.
:::

## DIA-NN workflow

### Part 1: Creating a spectral library

DIA-NN analysis consists of two steps. The first step is to generate a predicted spectral library for your experimental organism.

1. Download a sequence database from [UniProt](https://www.uniprot.org/uniprotkb?query=Human&facets=model_organism%3A9606%2Creviewed%3Atrue). 

    a. Go to https://www.uniprot.org/uniprotkb?query=*
    
    b. Select the **relevant organism** for your sample and **Reviewed (Swiss-Prot)** status to filter results.
    
    c. Click **Download**
    
    d. Change the format to **FASTA (canonical and isoform)**
    
    e. Select **No** under Compressed and download the Fasta file
    
:::callout

If you need to search for a variant protein or peptide that differs from the standard organism proteome, you can manually edit the protein sequences in the UniProt fasta in a text editor.

:::
    
2. Download a contaminant Fasta file from [here] (https://github.com/HaoGroup-ProtContLib/Protein-Contaminant-Libraries-for-DDA-and-DIA-Proteomics/tree/main). For most studies, the Universal Contaminants.fasta will be appropriate, but you may prefer to use one of the sample-type specific contaminant fastas depending on your experiment. More information on contaminants will be discussed in the next lesson.

3. Open the DIA-NN GUI or a command line interface. If there are multiple versions of DIA-NN installed on your device, ensure to update **DIA-NN exe** with the file path to the version you wish to use.

4. Add the UniProt and Contaminant fasta files you just downloaded. Examples are included below for the additional settings we recommend changing from the default. Depending on your experiment and research goals, you may wish to change additional settings from the default - read more on the [DIA-NN GitHub page](https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings)

:::warning

At the time of writing, DIA-NN 2.2.0 cannot read or write files saved in an external drive. Please ensure all input and output files are saved on a local drive to avoid error.

:::

5. Run. DIA-NN will output a predicted spectral library in the `.speclib` file type.


This spectral library can be re-used for all of analyses of samples from the same organism. It is recommended to re-download the UniProt fasta and re-run this step approximately once a month to ensure your library is up-to-date.


![Example of the DIA-NN GUI used to generate a predicted spectral library. Settings which have been altered from the default are highlighted in red](episodes/fig/02diann_DIANNgui_speclib.png)

```{bash eval=FALSE}
C:\DIA-NN\2.2.0\diann.exe ` # Replace with file path to DIA-NN on your device
--lib "" `
--out "C:\path\to\output\report.parquet" ` # Replace with file path to output folder
--out-lib "C:\path\to\output\report-lib.parquet" ` # Replace with file path to output folder
--fasta "C:\path\to\UniprotFasta\.fasta" ` # Replace with file path to Uniprot fasta
--fasta "C:\path\to\ContaminantsFasta\.fasta" ` # Replace with file path to contaminants fasta
--threads 30 ` # Replace with the number of threads to use for this analysis
--verbose 1 `
--qvalue 0.01 `
--matrices  `
--gen-spec-lib `
--predictor `
--reannotate `
--fasta-search `
--min-fr-mz 200 `
--max-fr-mz 1800 `
--met-excision `
--min-pep-len 7 `
--max-pep-len 30 `
--min-pr-mz 300 `
--max-pr-mz 1800 `
--min-pr-charge 1 `
--max-pr-charge 4 `
--cut K*,R* `
--missed-cleavages 1 `
--unimod4 `
--rt-profiling 
```

:::challenge

If you want to run DIA-NN from the command line but are unsure how to write the code, you can update your settings in the GUI, click 'Run', and copy the commands output in the log. That's how we got the example code above! Can you match each line of code to the corresponding setting in the GUI?

:::

<br>

### Part 2: Analysis

The second step is to search your raw files (mass spectrometry output files: .raw, .wiff, .mzML or .dia format) against the spectral library.

1. Open the DIA-NN GUI or a command line interface. If there are multiple versions of DIA-NN installed on your device, ensure to update **DIA-NN exe** with the file path to the version you wish to use.

2. Upload your raw files.

    - If using the GUI, under **Input** select **Raw** then select all of the files you wish to search
    - If using the command line, you can use `--f` to list every file or use `--dir [folder]` to instruct DIA-NN to process all raw files saved in a particular folder.

3. Upload your spectral library.

4. Adjust settings according to your experiment and research goals. Again, examples are included below for the additional settings we recommend changing from the default. You may wish to change additional settings from the default - read more on the [DIA-NN GitHub page](https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings)

    - In this example, the **protease** Trypsin was used to generate peptides during sample preparation.
    - Leaving **Mass accuracy** and **MS1 accuracy** on the default 0 means DIA-NN will optimise these parameters automatically for the first run in the experiment and then reuse the optimised settings for other runs. The DIA-NN GitHub recommends different settings depending on the  type of mass spectrometer used to generate your data. In this example, data were generated on a TripleTOF 6600, so we set Mass accuracy to 20.0 and MS1 accuracy to 12.0
    - Set the number of **Threads** to be no more than the number of logical cores on your computer.
    - In this experiment, we have set the **maximum number of variable modifications** to 0; however, it is common for this setting to be relaxed in the range of 1 - 3.
    - From the DIA-NN GitHub page: *MBR activates a mode in DIA-NN which we call 'match-between-runs'. Note that the algorithm is fundamentally different from MBR in some DDA software. In MBR mode, DIA-NN does two passes over the data. During the first pass, it creates an empirical spectral library from the data. During the second pass, it reanalyses the experiment with this empirical library, which may result in much improved identification numbers, data completeness and quantification accuracy. Always use MBR for quantitative analyses (as opposed to analyses just aimed at creating an empirical library), unless you are analysing with a DIA-based empirical library (in that case do not use MBR). If you are analysing with a project-specific DDA-based library, it may make sense to compare results with and without MBR. For this, analyse first with MBR. You will get both the final MBR report as well as a report based on the first MBR pass (named main_report_name-first-pass.parquet). Compare the identification numbers and data completeness in the two reports. If you decide the performance is higher for the first pass, just uncheck MBR, select Reuse .quant files, change the main output file name and click Run.*

:::callout

You may wish to generate an empirical spectral library from your data. This will be smaller than a predicted library and therefore increase the speed of analysis; however, may result in fewer identifications. We only recommend using an empirical library if generated from high quality data.

You can generate an empirical library by selecting **Generate spectral library** and specifying the location of its output.

:::

![Example of the DIA-NN GUI used to analyse the data described in this tutorial. Settings which have been altered from the default are highlighted in red](episodes/fig/02diann_DIANNgui_analysis.png)

```{bash eval=FALSE}
C:\DIA-NN\2.2.0\diann.exe ` # Replace with file path to DIA-NN on your device
--lib "" `
--out "C:\path\to\output\report.parquet" ` # Replace with file path to output folder
--out-lib "C:\path\to\output\report-lib.parquet" ` # Replace with file path to output folder
--fasta "C:\path\to\UniprotFasta\.fasta" ` # Replace with file path to Uniprot fasta
--fasta "C:\path\to\ContaminantsFasta\.fasta" ` # Replace with file path to contaminants fasta
--threads 30 ` # Replace with the number of threads to use for this analysis
--verbose 1 `
--qvalue 0.01 `
--matrices  `
--gen-spec-lib `
--predictor `
--reannotate `
--fasta-search `
--min-fr-mz 200 `
--max-fr-mz 1800 `
--met-excision `
--min-pep-len 7 `
--max-pep-len 30 `
--min-pr-mz 300 `
--max-pr-mz 1800 `
--min-pr-charge 1 `
--max-pr-charge 4 `
--cut K*,R* `
--missed-cleavages 1 `
--unimod4 `
--rt-profiling 
```

## DIA-NN output

DIA-NN output includes several files

## General Workflow

The DIA-NN workflow typically consists of the following steps:

1. **Prepare input data**
   - Raw DIA data files (`.raw`, `.wiff`, or converted `.mzML` files).
   - (Optional) A spectral library (`.tsv` or `.speclib`).

2. **Run DIA-NN**
   - Can be run via the **graphical user interface (GUI)** or **command line**.
   - Specify inputs, output directory, and parameters (e.g., FDR cutoff, protein grouping, library mode).

3. **Output files**
   - Quantification tables for proteins and peptides (`report.tsv` or `.parquet`).
   - Run summaries and performance metrics.
   - Optional normalized intensity data for downstream statistical analysis.

4. **Post-processing**
   - Further analysis can be done in R (e.g., `tidyverse`, `MSstats`, `limma`), Python, or visualization tools.
   - Data can also be imported into platforms like **Perseus**, **Spectronaut**, or **Skyline** for downstream exploration.



## Example Command-Line Usage

Below is an example of running DIA-NN in command-line mode (adjust paths for your setup):

```bash
DIA-NN.exe --f "path/to/mzML_files/" \
           --out "path/to/output_folder/" \
           --lib "path/to/library.tsv" \
           --threads 8 \
           --fasta "path/to/fasta_file.fasta" \
           --qvalue 0.01 \
           --verbose 1
```

This command tells DIA-NN to:

Analyze all .mzML files in a folder

Use a specified spectral library and FASTA file

Run on 8 threads

Output results with a 1% false discovery rate cutoff



:::spoiler
# More about DIANN

- For more background information, see the original publication:  
*Demichev, V., et al. (2020). DIA-NN: neural networks and interference correction enable deep proteome coverage in high throughput. Nature Methods.*  [https://doi.org/10.1038/s41592-019-0638-x](https://doi.org/10.1038/s41592-019-0638-x)

- For further details about the different DIA-NN settings, see the [GitHub page] (https://github.com/vdemichev/DiaNN?tab=readme-ov-file)

- If you come across any issues, the creators are fairly active at responding to [issues](https://github.com/vdemichev/DiaNN/issues) and [discussions](https://github.com/vdemichev/DiaNN/discussions) on their GitHub

:::

::::::::::::::::::::::::::::::::::::: keypoints 

- DIANN is fun and is theoretically easy to use!

::::::::::::::::::::::::::::::::::::::::::::::::



