---
title: 'Network and Enrichment Analysis'
teaching: 42
exercises: 3
---

::: questions
-   How can DEA results be used to identify biologically meaningful patterns?
-   How can we visualise and interpret protein–protein interaction (PPI) networks and pathway enrichment results?
:::

::: objectives
-   Visualise significant proteins using network analysis (STRING)
-   Identify enriched biological processes and pathways (GO and KEGG)
:::

```{r, results="hide", message=FALSE, warning = FALSE, echo=FALSE}
library(limpa)
library(readxl)
library(pheatmap)
library(EnhancedVolcano)
library(STRINGdb)
library(clusterProfiler)
library(org.Hs.eg.db)
library(arrow)
library(rpx)
library(dplyr)
library(stringr)

y.peptide <- readDIANN(file='data/MBIntroToProteomics.parquet',format="parquet",
                       q.columns = c("Q.Value","Lib.Q.Value","Lib.PG.Q.Value"),
                       q.cutoffs = 0.01)

samples_stool <- as.data.frame(readxl::read_excel('data/SampleAnnotation.xlsx'))

# Create sample_name column that matches peptide matrix column names
samples_stool$fixed_file_name <- gsub("\\\\", "/", samples_stool$file_name)
samples_stool$fixed_file_name <- basename(samples_stool$fixed_file_name)
samples_stool$sample_name <- gsub(".mzML", "",samples_stool$fixed_file_name)

# Filter to only samples present in our dataset
samples_stool <- samples_stool %>% filter(samples_stool$sample_name %in% colnames(y.peptide$E))

# Ensure sample row order matches column order of the peptide matrix
sample_info <- samples_stool[samples_stool$sample_name %in% colnames(y.peptide$E),]
rownames(sample_info) <- sample_info$sample_name
y.peptide$E <- y.peptide$E[,rownames(sample_info)]

# Add experimental metadata to EList
y.peptide$targets <- sample_info[, c('Class', 'Batch')]

y.peptide <- filterNonProteotypicPeptides(y.peptide)
y.peptide <- filterCompoundProteins(y.peptide)
y.peptide <- filterSingletonPeptides(y.peptide, min.n.peptides = 2)

# Remove contaminant proteins from the genes dataframe
y.peptide$genes <- y.peptide$genes %>% filter(!str_detect(Protein.Group, "Cont_"))

# Match peptides in the matrix to only those remaining in the dataframe
y.peptide$E <- y.peptide$E[rownames(y.peptide$genes), ]

dpcfit <- dpcON(y.peptide, robust=TRUE)

y.protein <- dpcQuant(y.peptide, "Protein.Names", dpc=dpcfit)

# Set colors for later plotting
Class <- factor(y.peptide$targets$Class)
Class.color <- Class
levels(Class.color) <- hcl.colors(nlevels(Class), palette = "cividis")

Batch <- factor(y.peptide$targets$Batch)
Batch.color <- Batch
levels(Batch.color) <- hcl.colors(nlevels(Batch), palette = "Set 2")

# Set the control group as our reference
Class <- relevel(Class, ref = "Ctrl")
## We also need to reset our colors to match the correct labels
Class.color <- Class
levels(Class.color) <- hcl.colors(nlevels(Class), palette = "cividis")

# Fit a linear model
design <- model.matrix(~Class + Batch)

fit <- dpcDE(y.protein, design, plot = FALSE)
fit <- eBayes(fit)

# Save our DEA results as a dataframe
results <- topTable(fit, coef = 2, number = Inf)

# Filter for the up to 50 most significant results
sig_proteins <- results %>%
  filter(adj.P.Val < 0.05) %>% # filter by p value
  top_n(50, wt = abs(logFC)) %>% # filter by absolute log fold change
  pull(Protein.Names)
expr_matrix <- y.protein$E[sig_proteins, ]

# Scale the data and visualise via heatmap
scaled_expr <- t(scale(t(expr_matrix)))
```

Now that we have a list of differentially expressed (DE) proteins, we can explore how these proteins interact with each other and what biological functions or pathways they are involved in.

This step helps translate statistical significance into biological meaning by integrating our results with known protein–protein interactions (PPIs) and functional annotation databases.


## STRING protein–protein interaction network analysis

The [STRING](https://string-db.org/) database integrates known and predicted PPIs from multiple sources (e.g. experimental data, text mining, prediction methods, and curated databases).

We can visualise DE proteins within the context of their **interaction network** to identify clusters or modules potentially corresponding to biological pathways.

```{r, eval=FALSE}
# Create an instance of the STRINGdb reference class for humans (species 9606)
string_db <- STRINGdb$new(version = "11.5", species = 9606, score_threshold = 400)

# Map STRING identifiers to gene identifiers in our DEA results dataframe
mapped <- string_db$map(results, "Protein.Names", removeUnmappedRows = TRUE)

# Only map the proteins of significance
de_proteins_mapped <- mapped %>% filter(adj.P.Val < 0.05)

# Plot network of significant proteins
string_db$plot_network(de_proteins_mapped$STRING_id)

```

*Note: if the above code takes too long to load, you can download an RData that contains the `string_db` and `mapped` functions [here](https://github.com/egmg726/intro-to-proteomics/blob/main/episodes/data/string_db.RData).*

```{r, echo=FALSE}

load("data/string_db.RData")

de_proteins_mapped <- mapped %>% filter(adj.P.Val < 0.05)
hits <- de_proteins_mapped$STRING_id

# Plot network of significant proteins
string_db$plot_network(hits)

```


:::: discussion

How many clusters are there? 

How many of these interactions are predicted versus experimentally validated? (See image below for legend)

::::


You can explore how these proteins interact in an interactive setting by going to [https://string-db.org/](https://string-db.org/) and further exploring your proteins of interest.

![Network plot for PRTN3 from string-db.org](episodes/fig/05enrichmentanalysis_stringdb.png)

:::callout

# More data exploration using STRINGdb

There are many ways of exploring your data using the STRING database - too many to cover in one tutorial! 

You can learn more by reading the [vignette](https://www.bioconductor.org/packages/release/bioc/vignettes/STRINGdb/inst/doc/STRINGdb.pdf) or inspecting available functions within the `STRINGdb` package by running:

```{r}

STRINGdb$methods()

```

:::

:::spoiler
# Read more about STRING

*Szklarczyk, D., Nastou, K., Koutrouli, M., Kirsch, R., Mehryary, F., Hachilif, R., Hu, D., Peluso, M. E., Huang, Q., Fang, T., Doncheva, N. T., Pyysalo, S., Bork, P., Jensen, L. J., & von Mering, C. (2025). The STRING database in 2025: protein networks with directionality of regulation. Nucleic acids research, 53(D1), D730–D737. https://doi.org/10.1093/nar/gkae1113*

:::


## Functional enrichment analysis (GO and KEGG)

Using the significant DE proteins, we can perform **functional enrichment analysis** to determine whether certain biological processes (BP), cellular components (CC), or molecular functions (MF) are over-represented.

Two common types of enrichment exploration are:

-   **Gene Ontology (GO)**: describes BPs, CCs, amd MFs
-   **KEGG pathways**: curated molecular interactions

### GO enrichment analysis

```{r}

#Subset genes from results table
de_proteins <- results %>% filter(adj.P.Val < 0.05)

# Convert UniProt IDs to Entrez IDs for enrichment analysis
converted <- bitr(de_proteins$Protein.Group,
                  fromType = "UNIPROT",
                  toType = "ENTREZID",
                  OrgDb = org.Hs.eg.db)

# GO Biological Process enrichment
ego <- enrichGO(gene = converted$ENTREZID,
                OrgDb = org.Hs.eg.db,
                ont = "ALL", # looking at all 3 subontologies
                pAdjustMethod = "BH",
                pvalueCutoff = 0.05,
                readable = TRUE)

# Show plot
dotplot(ego, showCategory = 10)


```

### KEGG pathway enrichment analysis

```{r}

# KEGG pathway enrichment
ekegg <- enrichKEGG(gene = converted$ENTREZID,
                    organism = 'hsa',
                    pvalueCutoff = 0.05)

dotplot(ekegg, showCategory = 10)

```


:::: challenge
How do these figures compare to [Figure 4](https://mdpi-res.com/biomedicines/biomedicines-12-00333/article_deploy/html/images/biomedicines-12-00333-g004.png) in the original paper?

::: solution
While there are many similar pathways, the exact pathways are not identical. This is likely due to the fact we are using a subset of the larger dataset in this tutorial.
:::
::::



::: keypoints
-   Differential expression analysis identifies statistically significant changes in protein abundance between conditions, but does not necessarily tell us the biological significance of such identified proteins.
-   Network analysis (STRING) can reveal physical or functional interactions among DE proteins.
-   Enrichment analysis (GO/KEGG) can link those proteins to known biological processes or pathways.
-   Using multiple approaches can help you interpret proteomics data to measure reliability and consistency of results.
:::
