---
title: 'Network and Enrichment Analysis'
teaching: 10
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can DE results be used to identify biologically meaningful patterns?  
- How can we visualise and interpret protein–protein interaction (PPI) networks and pathway enrichment results?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- To visualise significant proteins using network analysis (STRING)  
- To identify enriched biological processes and pathways (GO and KEGG)  

::::::::::::::::::::::::::::::::::::::::::::::::

```{r, results="hide", message=FALSE, warning = FALSE, echo=FALSE}
library(limpa)
library(readxl)
library(pheatmap)
library(EnhancedVolcano)
library(STRINGdb)
library(clusterProfiler)
library(org.Hs.eg.db)
library(arrow)
library(rpx)
library(dplyr)
library(stringr)

y.peptide <- readDIANN(file='data/MBIntroToProteomics.parquet',format="parquet",
                       q.columns = c("Q.Value","Lib.Q.Value","Lib.PG.Q.Value"),
                       q.cutoffs = 0.01)

samples_stool <- as.data.frame(readxl::read_excel('data/SampleAnnotation.xlsx'))
head(samples_stool)

# Create sample_name column that matches peptide matrix column names
samples_stool$fixed_file_name <- gsub("\\\\", "/", samples_stool$file_name)
samples_stool$fixed_file_name <- basename(samples_stool$fixed_file_name)
samples_stool$sample_name <- gsub(".mzML", "",samples_stool$fixed_file_name)

# Filter to only samples present in our dataset
samples_stool <- samples_stool %>% filter(samples_stool$sample_name %in% colnames(y.peptide$E))

# Ensure sample row order matches column order of the peptide matrix
sample_info <- samples_stool[samples_stool$sample_name %in% colnames(y.peptide$E),]
rownames(sample_info) <- sample_info$sample_name
y.peptide$E <- y.peptide$E[,rownames(sample_info)]

# Add experimental metadata to EList
y.peptide$targets <- sample_info[, c('Class', 'Batch')]

# Set colors for later plotting
Class <- factor(y.peptide$targets$Class)
Class.color <- Class
levels(Class.color) <- hcl.colors(nlevels(Class), palette = "cividis")

Batch <- factor(y.peptide$targets$Batch)
Batch.color <- Batch
levels(Batch.color) <- hcl.colors(nlevels(Batch), palette = "Set 2")

y.peptide <- filterNonProteotypicPeptides(y.peptide)
y.peptide <- filterCompoundProteins(y.peptide)
y.peptide <- filterSingletonPeptides(y.peptide, min.n.peptides = 2)

# Remove contaminant proteins from the genes dataframe
y.peptide$genes <- y.peptide$genes %>% filter(!str_detect(Protein.Group, "Cont_"))

# Match peptides in the matrix to only those remaining in the dataframe
y.peptide$E <- y.peptide$E[rownames(y.peptide$genes), ]

dpcfit <- dpcON(y.peptide, robust=TRUE)



y.protein <- dpcQuant(y.peptide, "Protein.Names", dpc=dpcfit)

# Set the control group as our reference
Class <- relevel(Class, ref = "Ctrl")
## We also need to reset our colors to match the correct labels
Class.color <- Class
levels(Class.color) <- hcl.colors(nlevels(Class), palette = "cividis")

# Fit a linear model
design <- model.matrix(~Class + Batch)

fit <- dpcDE(y.protein, design, plot = FALSE)
fit <- eBayes(fit)

# Save our DEA results as a dataframe
results <- topTable(fit, coef = 2, number = Inf)

# Filter for the up to 50 most significant results
sig_proteins <- results %>%
  filter(adj.P.Val < 0.05) %>% # filter by p value
  top_n(50, wt = abs(logFC)) %>% # filter by absolute log fold change
  pull(Protein.Names)
expr_matrix <- y.protein$E[sig_proteins, ]

# Scale the data and visualise via heatmap
scaled_expr <- t(scale(t(expr_matrix)))



```


## Network and Enrichment Analysis

Now that we have a list of differentially expressed proteins, we can explore how these proteins interact with each other and what biological functions or pathways they are involved in.

This step helps translate statistical significance into biological meaning by integrating our results with known protein–protein interactions (PPI) and functional annotation databases.

### STRING Protein–Protein Interaction Network

The STRING database integrates known and predicted protein–protein interactions from multiple sources (e.g., experimental data, text mining, co-expression, and curated databases).

We can visualise DE proteins within the context of their interaction network to identify clusters or modules potentially corresponding to biological pathways.

```{r, eval=FALSE}

string_db <- STRINGdb$new(version = "11.5", species = 9606, score_threshold = 400)
mapped <- string_db$map(results, "Protein.Names", removeUnmappedRows = TRUE)
de_proteins_mapped <- mapped %>% filter(adj.P.Val < 0.05)

# Plot network of significant proteins
string_db$plot_network(de_proteins_mapped$STRING_id)

```


Note if not working, use this:

```{r}

load("data/string_db.RData")

de_proteins_mapped <- mapped %>% filter(adj.P.Val < 0.05)

hits <- de_proteins_mapped$STRING_id

# Plot network of significant proteins
string_db$plot_network(hits)


```

You can inspect available functions within the STRINGdb package by running:

```{r}

STRINGdb$methods()

```



### Functional Enrichment (GO and KEGG)

Using the identified significant DE proteins, we can perform functional enrichment analysis to determine whether certain biological processes (BP), cellular components (CC), or molecular functions (MF) are over-represented.

Two common types of enrichment are:

- Gene Ontology (GO): describes BPs, CCs, amd MFs
- KEGG pathways: provides curated biochemical pathways



```{r}

#Subset genes from results table
de_proteins <- results %>% filter(adj.P.Val < 0.05)

# Convert UniProt IDs to Entrez IDs for enrichment analysis
converted <- bitr(de_proteins$Protein.Group,
                  fromType = "UNIPROT",
                  toType = "ENTREZID",
                  OrgDb = org.Hs.eg.db)

# GO Biological Process enrichment
ego <- enrichGO(gene = converted$ENTREZID,
                OrgDb = org.Hs.eg.db,
                ont = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff = 0.05,
                readable = TRUE)

dotplot(ego, showCategory = 10)

# KEGG pathway enrichment
ekegg <- enrichKEGG(gene = converted$ENTREZID,
                    organism = 'hsa',
                    pvalueCutoff = 0.05)

dotplot(ekegg, showCategory = 10)


```

:::challenge

How does this figure compare to [Figure 4](https://mdpi-res.com/biomedicines/biomedicines-12-00333/article_deploy/html/images/biomedicines-12-00333-g004.png) in the original paper?

:::solution

While there are many immune-related pathways in both, the exact pathways are not identical. This is likely due to the fact we are using a subset of the larger dataset.




:::
:::::




::::::::::::::::::::::::::::::::::::: keypoints 

- DE analysis identifies statistically significant changes in protein abundance between conditions.
- Network analysis (STRING) reveals potential physical or functional interactions among DE proteins.
- Enrichment analysis (GO/KEGG) links those proteins to known biological processes or pathways.
- Using multiple approaches can help you interpret proteomics data in a systems biology context.

::::::::::::::::::::::::::::::::::::::::::::::::

